{
    "items": [
        {
            "tags": [
                "ios",
                "swift"
            ],
            "owner": {
                "reputation": 13688,
                "user_id": 2223390,
                "user_type": "registered",
                "accept_rate": 67,
                "profile_image": "https://i.stack.imgur.com/viIH0.jpg?s=128&g=1",
                "display_name": "Mani",
                "link": "https://stackoverflow.com/users/2223390/mani"
            },
            "is_answered": true,
            "view_count": 5663,
            "accepted_answer_id": 31946217,
            "answer_count": 4,
            "score": 8,
            "last_activity_date": 1526415924,
            "creation_date": 1401947448,
            "question_id": 24052545,
            "body_markdown": "I&#39;ve just opened existing project in xcode 6 beta. But it&#39;s crash with following error message.\r\n\r\n   \r\n\r\n    dyld: problem loading iOS simulator dyld\r\n    2014-06-05 11:15:06.604 ibtoold[952:507] Init failed: Error Domain=IBMessageChannelErrorDomain Code=1 &quot;Failed to communicate with Interface Builder&quot; UserInfo=0x7fd09861dbf0 {NSLocalizedDescription=Failed to communicate with Interface Builder, NSLocalizedFailureReason=Interface Builder Cocoa Touch Tool crashed: EXC_BREAKPOINT (SIGTRAP), IBCrashLog=\r\n    \r\n    Dyld Error Message:\r\n      problem loading iOS simulator dyld\r\n    }\r\n    2014-06-05 11:15:06.605 ibtoold[952:507] [MT] DVTAssertions: Warning in /SourceCache/IDEInterfaceBuilder/IDEInterfaceBuilder-6154.17/InterfaceBuilderKit/Utilities/IBAbstractMessageChannelInterfaceBuilderToolManager.m:76\r\n    Details:  Failed to attach to Interface Builder Cocoa Touch Tool with error: Error Domain=com.apple.InterfaceBuilder Code=-1 &quot;Encountered an error communicating with Interface Builder Cocoa Touch Tool.&quot; UserInfo=0x7fd09b3a1360 {NSUnderlyingError=0x7fd09861d710 &quot;Failed to communicate with Interface Builder&quot;, NSLocalizedDescription=Encountered an error communicating with Interface Builder Cocoa Touch Tool., NSLocalizedRecoverySuggestion=Please check Console.app for crash reports for &quot;Interface Builder Cocoa Touch Tool&quot; for further information., NSLocalizedFailureReason=Interface Builder Cocoa Touch Tool (962) failed to launch and exited with status (null), signal 5.\r\n    \r\n    \r\n    \r\n    Dyld Error Message:\r\n      problem loading iOS simulator dyld\r\n    }\r\n    Object:   &lt;IBCocoaTouchToolManager: 0x7fd09b478a40&gt;\r\n    Method:   -attachToToolForTargetRuntime:scaleFactor:assertOnFailureToAttach:assertOnPostLaunchRequestFailures:error:\r\n    Thread:   &lt;NSThread: 0x7fd098614780&gt;{name = (null), num = 1}\r\n    Please file a bug at http://bugreport.apple.com with this warning message and any useful information you can provide.\r\n    2014-06-05 11:15:06.914 ibtoold[952:507] [MT] DVTAssertions: ASSERTION FAILURE in /SourceCache/IDEInterfaceBuilderCocoaTouch/IDEInterfaceBuilderCocoaTouch-6153.11/InterfaceBuilder/Utilities/IBObjectMarshalling.m:45\r\n    Details:  Failed to launch Interface Builder Cocoa Touch Tool: Error Domain=com.apple.InterfaceBuilder Code=-1 &quot;Encountered an error communicating with Interface Builder Cocoa Touch Tool.&quot; UserInfo=0x7fd09b3a1360 {NSUnderlyingError=0x7fd09861d710 &quot;Failed to communicate with Interface Builder&quot;, NSLocalizedDescription=Encountered an error communicating with Interface Builder Cocoa Touch Tool., NSLocalizedRecoverySuggestion=Please check Console.app for crash reports for &quot;Interface Builder Cocoa Touch Tool&quot; for further information., NSLocalizedFailureReason=Interface Builder Cocoa Touch Tool (962) failed to launch and exited with status (null), signal 5.\r\n\r\n\r\nIs any workaround for xib to run on xcode beta? ",
            "link": "https://stackoverflow.com/questions/24052545/swift-failed-to-communicate-with-interface-builder",
            "title": "swift : Failed to communicate with Interface Builder",
            "body": "<p>I've just opened existing project in xcode 6 beta. But it's crash with following error message.</p>\n\n<pre><code>dyld: problem loading iOS simulator dyld\n2014-06-05 11:15:06.604 ibtoold[952:507] Init failed: Error Domain=IBMessageChannelErrorDomain Code=1 \"Failed to communicate with Interface Builder\" UserInfo=0x7fd09861dbf0 {NSLocalizedDescription=Failed to communicate with Interface Builder, NSLocalizedFailureReason=Interface Builder Cocoa Touch Tool crashed: EXC_BREAKPOINT (SIGTRAP), IBCrashLog=\n\nDyld Error Message:\n  problem loading iOS simulator dyld\n}\n2014-06-05 11:15:06.605 ibtoold[952:507] [MT] DVTAssertions: Warning in /SourceCache/IDEInterfaceBuilder/IDEInterfaceBuilder-6154.17/InterfaceBuilderKit/Utilities/IBAbstractMessageChannelInterfaceBuilderToolManager.m:76\nDetails:  Failed to attach to Interface Builder Cocoa Touch Tool with error: Error Domain=com.apple.InterfaceBuilder Code=-1 \"Encountered an error communicating with Interface Builder Cocoa Touch Tool.\" UserInfo=0x7fd09b3a1360 {NSUnderlyingError=0x7fd09861d710 \"Failed to communicate with Interface Builder\", NSLocalizedDescription=Encountered an error communicating with Interface Builder Cocoa Touch Tool., NSLocalizedRecoverySuggestion=Please check Console.app for crash reports for \"Interface Builder Cocoa Touch Tool\" for further information., NSLocalizedFailureReason=Interface Builder Cocoa Touch Tool (962) failed to launch and exited with status (null), signal 5.\n\n\n\nDyld Error Message:\n  problem loading iOS simulator dyld\n}\nObject:   &lt;IBCocoaTouchToolManager: 0x7fd09b478a40&gt;\nMethod:   -attachToToolForTargetRuntime:scaleFactor:assertOnFailureToAttach:assertOnPostLaunchRequestFailures:error:\nThread:   &lt;NSThread: 0x7fd098614780&gt;{name = (null), num = 1}\nPlease file a bug at http://bugreport.apple.com with this warning message and any useful information you can provide.\n2014-06-05 11:15:06.914 ibtoold[952:507] [MT] DVTAssertions: ASSERTION FAILURE in /SourceCache/IDEInterfaceBuilderCocoaTouch/IDEInterfaceBuilderCocoaTouch-6153.11/InterfaceBuilder/Utilities/IBObjectMarshalling.m:45\nDetails:  Failed to launch Interface Builder Cocoa Touch Tool: Error Domain=com.apple.InterfaceBuilder Code=-1 \"Encountered an error communicating with Interface Builder Cocoa Touch Tool.\" UserInfo=0x7fd09b3a1360 {NSUnderlyingError=0x7fd09861d710 \"Failed to communicate with Interface Builder\", NSLocalizedDescription=Encountered an error communicating with Interface Builder Cocoa Touch Tool., NSLocalizedRecoverySuggestion=Please check Console.app for crash reports for \"Interface Builder Cocoa Touch Tool\" for further information., NSLocalizedFailureReason=Interface Builder Cocoa Touch Tool (962) failed to launch and exited with status (null), signal 5.\n</code></pre>\n\n<p>Is any workaround for xib to run on xcode beta? </p>\n"
        },
        {
            "tags": [
                "django",
                "post",
                "save"
            ],
            "owner": {
                "reputation": 46,
                "user_id": 1397243,
                "user_type": "registered",
                "accept_rate": 0,
                "profile_image": "https://www.gravatar.com/avatar/90111a69228d45296f713c0c0a8f0e44?s=128&d=identicon&r=PG",
                "display_name": "Bob Donahue",
                "link": "https://stackoverflow.com/users/1397243/bob-donahue"
            },
            "is_answered": false,
            "view_count": 4,
            "answer_count": 0,
            "score": 0,
            "last_activity_date": 1526415922,
            "creation_date": 1526415922,
            "question_id": 50358654,
            "body_markdown": "I have a site that updates data from a call to an API.\r\nSo, in order to poll the API for the relevant fields, I have an &quot;ingest_on_save&quot; flag on the model, that&#39;s checked in a @pre_save handler, and after grabbing the API data will create or update the record.   At the end of the @pre_save it turns the ingest_on_save to False and proceeds with save().\r\n\r\nIt works great!\r\n\r\nBUT the API also has pointers to child objects that are in a different model (e.g., a Book models with a FK to Author).   \r\n\r\nIf I change my Author model to also have a &quot;ingest_books&quot; flag, I can create/edit Books from it&#39;s @pre_save similarly (from the Author @post_save - necessary because I need the PK if the Author is newly-created to set the FK between Author and Book), but I can&#39;t figure out how to unset the Author.ingest_books since save() has already been called (and calling it from it&#39;s @post_save creates an infinite recursion loop).\r\n\r\nI&#39;ve seen answers for other similar situations saying to use queryset.filter(pk=instance.pk).update except that there&#39;s no queryset that&#39;s reachable from @post_save (and those answers don&#39;t say how they were able to reach it)...\r\n\r\n... ideas?\r\n",
            "link": "https://stackoverflow.com/questions/50358654/django-creating-child-objects-within-post-save",
            "title": "Django: creating child objects within post_save",
            "body": "<p>I have a site that updates data from a call to an API.\nSo, in order to poll the API for the relevant fields, I have an \"ingest_on_save\" flag on the model, that's checked in a @pre_save handler, and after grabbing the API data will create or update the record.   At the end of the @pre_save it turns the ingest_on_save to False and proceeds with save().</p>\n\n<p>It works great!</p>\n\n<p>BUT the API also has pointers to child objects that are in a different model (e.g., a Book models with a FK to Author).   </p>\n\n<p>If I change my Author model to also have a \"ingest_books\" flag, I can create/edit Books from it's @pre_save similarly (from the Author @post_save - necessary because I need the PK if the Author is newly-created to set the FK between Author and Book), but I can't figure out how to unset the Author.ingest_books since save() has already been called (and calling it from it's @post_save creates an infinite recursion loop).</p>\n\n<p>I've seen answers for other similar situations saying to use queryset.filter(pk=instance.pk).update except that there's no queryset that's reachable from @post_save (and those answers don't say how they were able to reach it)...</p>\n\n<p>... ideas?</p>\n"
        },
        {
            "tags": [
                "shell",
                "unix",
                "sh"
            ],
            "owner": {
                "reputation": 5797,
                "user_id": 661624,
                "user_type": "registered",
                "accept_rate": 72,
                "profile_image": "https://www.gravatar.com/avatar/347231f8387561ef324e25b48b96ba16?s=128&d=identicon&r=PG&f=1",
                "display_name": "Henley Chiu",
                "link": "https://stackoverflow.com/users/661624/henley-chiu"
            },
            "is_answered": false,
            "view_count": 25,
            "answer_count": 0,
            "score": 0,
            "last_activity_date": 1526415910,
            "creation_date": 1526414724,
            "last_edit_date": 1526415910,
            "question_id": 50358395,
            "body_markdown": "I have a rsync command that does this: \r\n\r\n    rsync -a path postgres@12.19.232.116:/home/postgres/data/wals &lt; /dev/null\r\n\r\nI want to also redirect the output of this command to a file (ie &gt; output.log), while preserving &lt; /dev/null. How do I do so?\r\n",
            "link": "https://stackoverflow.com/questions/50358395/how-to-combine-unix-input-output-redirects",
            "title": "How to combine Unix input + output redirects?",
            "body": "<p>I have a rsync command that does this: </p>\n\n<pre><code>rsync -a path postgres@12.19.232.116:/home/postgres/data/wals &lt; /dev/null\n</code></pre>\n\n<p>I want to also redirect the output of this command to a file (ie > output.log), while preserving &lt; /dev/null. How do I do so?</p>\n"
        },
        {
            "tags": [
                "sql",
                "sql-server"
            ],
            "owner": {
                "reputation": 89,
                "user_id": 369323,
                "user_type": "registered",
                "accept_rate": 70,
                "profile_image": "https://www.gravatar.com/avatar/3efd4431e2e5e1ef9958c45bb9ccf08d?s=128&d=identicon&r=PG",
                "display_name": "Tom",
                "link": "https://stackoverflow.com/users/369323/tom"
            },
            "is_answered": true,
            "view_count": 35,
            "answer_count": 2,
            "score": 0,
            "last_activity_date": 1526415897,
            "creation_date": 1526413869,
            "last_edit_date": 1526415593,
            "question_id": 50358190,
            "body_markdown": "I have a query that consists of 1 table and 2 sub queries. The table being a listing of all customers, 1 sub query is a listing all of the quotes given over a period of time for customers and the other sub query is a listing of all of the orders booked for a customer over the same period of time. What I am trying to do is return a result set that is a customer, the number of quotes given, and the number of orders booked over a given period of time. However what I am returning is only a listening of customers over the period of time that have an equivalent quote and order count. I feel like I am missing something obvious within the context of the query but I am unable to figure it out. Any help would be appreciated. Thank you.\r\n\r\nResult Set should look like this\r\n\r\n\r\nCustomer-------Quotes-------Orders Placed\r\n\r\naaa----------------4----------------4\r\n\r\nbbb----------------9----------------18\r\n\r\nccc----------------18----------------9\r\n\r\n\r\n    select \r\n    \t[Customer2].[Name] as [Customer2_Name],\r\n    \t(count( Quotes.UD03_Key3 )) as [Calculated_CustomerQuotes],\r\n    \t(count( Customer_Bookings.OrderHed_OrderNum )) as [Calculated_CustomerBookings]\r\n    from Erp.Customer as Customer2\r\n    left join  (select \r\n    \t[UD03].[Key3] as [UD03_Key3],\r\n    \t[UD03].[Key4] as [UD03_Key4],\r\n    \t[UD03].[Key1] as [UD03_Key1],\r\n    \t[UD03].[Date02] as [UD03_Date02]\r\n    from Ice.UD03 as UD03\r\n    inner join Ice.UD02 as UD02 on \r\n    \tUD03.Company = UD02.Company\r\n    And\r\n    \tCAST(CAST(UD03.Number09 AS INT) AS VARCHAR(30)) = UD02.Key1\r\n    \r\n    left outer join Erp.Customer as Customer on \r\n    \tUD03.Company = Customer.Company\r\n    And\r\n    \tUD03.Key1 = Customer.Name\r\n    \r\n    left outer join Erp.SalesTer as SalesTer on \r\n    \tCustomer.Company = SalesTer.Company\r\n    And\r\n    \tCustomer.TerritoryID = SalesTer.TerritoryID\r\n    \r\n    left outer join Erp.CustGrup as CustGrup on \r\n    \tCustomer.Company = CustGrup.Company\r\n    And\r\n    \tCustomer.GroupCode = CustGrup.GroupCode\r\n    \r\n     where (UD03.Key3 &lt;&gt; &#39;0&#39;))  as Quotes on \r\n    \tCustomer2.Name = Quotes.UD03_Key1\r\n    \r\n    left join  (select \r\n    \t[Customer1].[Name] as [Customer1_Name],\r\n    \t[OrderHed].[OrderNum] as [OrderHed_OrderNum],\r\n    \t[OrderDtl].[OrderLine] as [OrderDtl_OrderLine],\r\n    \t[OrderHed].[OrderDate] as [OrderHed_OrderDate]\r\n    from Erp.OrderHed as OrderHed\r\n    inner join Erp.Customer as Customer1 on \r\n    \tOrderHed.Company = Customer1.Company\r\n    And\r\n    \tOrderHed.BTCustNum = Customer1.CustNum\r\n    \r\n    inner join Erp.OrderDtl as OrderDtl on \r\n    \tOrderHed.Company = OrderDtl.Company\r\n    And\r\n    \tOrderHed.OrderNum = OrderDtl.OrderNum)  as Customer_Bookings on \r\n    \tCustomer2.Name = Customer_Bookings.Customer1_Name\r\n    \r\n     where Quotes.UD03_Date02 &gt;= &#39;5/15/2018&#39;  and Quotes.UD03_Date02 &lt;= &#39;5/15/2018&#39;  and Customer_Bookings.OrderHed_OrderDate &gt;=&#39;5/15/2018&#39;  and Customer_Bookings.OrderHed_OrderDate &lt;= &#39;5/15/2018&#39;\r\n    \r\n    group by [Customer2].[Name]",
            "link": "https://stackoverflow.com/questions/50358190/sql-query-with-counts-only-returning-equivalent-counts",
            "title": "SQL Query with counts only returning equivalent counts",
            "body": "<p>I have a query that consists of 1 table and 2 sub queries. The table being a listing of all customers, 1 sub query is a listing all of the quotes given over a period of time for customers and the other sub query is a listing of all of the orders booked for a customer over the same period of time. What I am trying to do is return a result set that is a customer, the number of quotes given, and the number of orders booked over a given period of time. However what I am returning is only a listening of customers over the period of time that have an equivalent quote and order count. I feel like I am missing something obvious within the context of the query but I am unable to figure it out. Any help would be appreciated. Thank you.</p>\n\n<p>Result Set should look like this</p>\n\n<p>Customer-------Quotes-------Orders Placed</p>\n\n<p>aaa----------------4----------------4</p>\n\n<p>bbb----------------9----------------18</p>\n\n<p>ccc----------------18----------------9</p>\n\n<pre><code>select \n    [Customer2].[Name] as [Customer2_Name],\n    (count( Quotes.UD03_Key3 )) as [Calculated_CustomerQuotes],\n    (count( Customer_Bookings.OrderHed_OrderNum )) as [Calculated_CustomerBookings]\nfrom Erp.Customer as Customer2\nleft join  (select \n    [UD03].[Key3] as [UD03_Key3],\n    [UD03].[Key4] as [UD03_Key4],\n    [UD03].[Key1] as [UD03_Key1],\n    [UD03].[Date02] as [UD03_Date02]\nfrom Ice.UD03 as UD03\ninner join Ice.UD02 as UD02 on \n    UD03.Company = UD02.Company\nAnd\n    CAST(CAST(UD03.Number09 AS INT) AS VARCHAR(30)) = UD02.Key1\n\nleft outer join Erp.Customer as Customer on \n    UD03.Company = Customer.Company\nAnd\n    UD03.Key1 = Customer.Name\n\nleft outer join Erp.SalesTer as SalesTer on \n    Customer.Company = SalesTer.Company\nAnd\n    Customer.TerritoryID = SalesTer.TerritoryID\n\nleft outer join Erp.CustGrup as CustGrup on \n    Customer.Company = CustGrup.Company\nAnd\n    Customer.GroupCode = CustGrup.GroupCode\n\n where (UD03.Key3 &lt;&gt; '0'))  as Quotes on \n    Customer2.Name = Quotes.UD03_Key1\n\nleft join  (select \n    [Customer1].[Name] as [Customer1_Name],\n    [OrderHed].[OrderNum] as [OrderHed_OrderNum],\n    [OrderDtl].[OrderLine] as [OrderDtl_OrderLine],\n    [OrderHed].[OrderDate] as [OrderHed_OrderDate]\nfrom Erp.OrderHed as OrderHed\ninner join Erp.Customer as Customer1 on \n    OrderHed.Company = Customer1.Company\nAnd\n    OrderHed.BTCustNum = Customer1.CustNum\n\ninner join Erp.OrderDtl as OrderDtl on \n    OrderHed.Company = OrderDtl.Company\nAnd\n    OrderHed.OrderNum = OrderDtl.OrderNum)  as Customer_Bookings on \n    Customer2.Name = Customer_Bookings.Customer1_Name\n\n where Quotes.UD03_Date02 &gt;= '5/15/2018'  and Quotes.UD03_Date02 &lt;= '5/15/2018'  and Customer_Bookings.OrderHed_OrderDate &gt;='5/15/2018'  and Customer_Bookings.OrderHed_OrderDate &lt;= '5/15/2018'\n\ngroup by [Customer2].[Name]\n</code></pre>\n"
        },
        {
            "tags": [
                "open-source",
                "diagram",
                "dia"
            ],
            "owner": {
                "reputation": 189,
                "user_id": 1655878,
                "user_type": "registered",
                "accept_rate": 50,
                "profile_image": "https://i.stack.imgur.com/3FRYH.jpg?s=128&g=1",
                "display_name": "argentum2f",
                "link": "https://stackoverflow.com/users/1655878/argentum2f"
            },
            "is_answered": false,
            "view_count": 15,
            "answer_count": 0,
            "score": 0,
            "last_activity_date": 1526415896,
            "creation_date": 1526415896,
            "question_id": 50358648,
            "body_markdown": "As far as I can tell there hasn&#39;t been any releases in 5+ years. Is it a dead project, or is there a fork of it somewhere with ongoing development?\r\n\r\nIt&#39;s a decent piece of software, but it&#39;s still missing at least one must have (to me) feature: the ability to rotate objects.",
            "link": "https://stackoverflow.com/questions/50358648/is-development-of-dia-diagram-software-permanently-halted",
            "title": "Is development of Dia (diagram software) permanently halted?",
            "body": "<p>As far as I can tell there hasn't been any releases in 5+ years. Is it a dead project, or is there a fork of it somewhere with ongoing development?</p>\n\n<p>It's a decent piece of software, but it's still missing at least one must have (to me) feature: the ability to rotate objects.</p>\n"
        },
        {
            "tags": [
                "docker"
            ],
            "owner": {
                "reputation": 310,
                "user_id": 2569618,
                "user_type": "registered",
                "accept_rate": 57,
                "profile_image": "https://www.gravatar.com/avatar/7c9333cbab7b8b700d2ad747d51feb81?s=128&d=identicon&r=PG",
                "display_name": "user2569618",
                "link": "https://stackoverflow.com/users/2569618/user2569618"
            },
            "is_answered": false,
            "view_count": 14,
            "answer_count": 0,
            "score": 0,
            "last_activity_date": 1526415886,
            "creation_date": 1526415886,
            "question_id": 50358645,
            "body_markdown": "I have a docker image that I created:  \r\n   \r\n    docker save dockerlibrary.lab/centos7-java8 &gt; latest.tar\r\n\r\nscp&#39;d the image to a secure zone and then loaded the image: \r\n    \r\n    docker load --input latest.tar\r\n\r\nHowever, I when I try to run the image, I get the following:\r\n\r\n    $ docker run -it dockerlibrary.lab/centos7-java8\r\n    Could not load host key: /etc/ssh/ssh_host_ecdsa_key\r\n    Could not load host key: /etc/ssh/ssh_host_ed25519_key\r\n\r\nAnd then docker hangs and I have to kill the process.  What did I miss in the instructions?\r\n\r\nSource machine:\r\n\r\n    # cat /etc/system-release\r\n    Red Hat Enterprise Linux Server release 6.6 (Santiago)\r\n    # docker --version\r\n    Docker version 1.7.1, build 786b29d/1.7.1\r\n\r\nTarget machine:\r\n\r\n    # cat /etc/system-release\r\n    CentOS Linux release 7.3.1611 (Core) \r\n    # docker --version\r\n    Docker version 18.03.1-ce, build 9ee9f40\r\n\r\n",
            "link": "https://stackoverflow.com/questions/50358645/docker-save-load-image-give-ssh-errors",
            "title": "docker save/load image give ssh errors",
            "body": "<p>I have a docker image that I created:  </p>\n\n<pre><code>docker save dockerlibrary.lab/centos7-java8 &gt; latest.tar\n</code></pre>\n\n<p>scp'd the image to a secure zone and then loaded the image: </p>\n\n<pre><code>docker load --input latest.tar\n</code></pre>\n\n<p>However, I when I try to run the image, I get the following:</p>\n\n<pre><code>$ docker run -it dockerlibrary.lab/centos7-java8\nCould not load host key: /etc/ssh/ssh_host_ecdsa_key\nCould not load host key: /etc/ssh/ssh_host_ed25519_key\n</code></pre>\n\n<p>And then docker hangs and I have to kill the process.  What did I miss in the instructions?</p>\n\n<p>Source machine:</p>\n\n<pre><code># cat /etc/system-release\nRed Hat Enterprise Linux Server release 6.6 (Santiago)\n# docker --version\nDocker version 1.7.1, build 786b29d/1.7.1\n</code></pre>\n\n<p>Target machine:</p>\n\n<pre><code># cat /etc/system-release\nCentOS Linux release 7.3.1611 (Core) \n# docker --version\nDocker version 18.03.1-ce, build 9ee9f40\n</code></pre>\n"
        },
        {
            "tags": [
                "cassandra",
                "kotlin",
                "datastax"
            ],
            "owner": {
                "reputation": 12203,
                "user_id": 1261825,
                "user_type": "registered",
                "accept_rate": 74,
                "profile_image": "https://www.gravatar.com/avatar/44eaaa3783922a617281e96bfe3f5343?s=128&d=identicon&r=PG",
                "display_name": "samanime",
                "link": "https://stackoverflow.com/users/1261825/samanime"
            },
            "is_answered": false,
            "view_count": 10,
            "answer_count": 0,
            "score": 0,
            "last_activity_date": 1526415881,
            "creation_date": 1526415881,
            "question_id": 50358644,
            "body_markdown": "I&#39;m using Kotlin with the DataStax Cassandra driver.\r\n\r\nWhen I call `Session::execute()` with a map, it seems like there might be some ambiguity because I get the `InvalidQueryException: Invalid amount of bind variables` exception.\r\n\r\nMy query had 5 named parameters, something like this:\r\n\r\n    private val QUERY = INSERT INTO my_table VALUES (:a, :b, :c, :d, :e)\r\n\r\nI was trying to execute against it like this:\r\n\r\n    session.execute(QUERY, mapOf(&quot;a&quot; to 1, &quot;b&quot; to 2, &quot;c&quot; to 3, &quot;d&quot; to 4, &quot;e&quot; to 5))\r\n\r\nSince I&#39;m certain that I had the correct number of parameters, and that they were all named correctly (and even in the same order), my best guess is it was instead using the other version of the function, which accepts `varargs`.\r\n\r\nThe expected signature of the second parameter is `Map&lt;String, Object&gt;`, or translated by Kotlin to `Map&lt;String!, Object!&gt;!`. I tried to cast it to the following:\r\n\r\n- `Map&lt;String, Object&gt;`\r\n- `Map&lt;String?, Object?&gt;`\r\n- `Map&lt;String?, Object?&gt;?`\r\n- `Map&lt;String, Any&gt;`\r\n- `Map&lt;String?, Any?&gt;`\r\n- `Map&lt;String?, Any?&gt;?`\r\n\r\nAnd none of those seemed to work.\r\n\r\nHow can I call `Session::execute()` with a `Map` as the second parameter?",
            "link": "https://stackoverflow.com/questions/50358644/datastax-session-ambiguous-execute-with-a-map",
            "title": "DataStax Session - Ambiguous execute() with a map",
            "body": "<p>I'm using Kotlin with the DataStax Cassandra driver.</p>\n\n<p>When I call <code>Session::execute()</code> with a map, it seems like there might be some ambiguity because I get the <code>InvalidQueryException: Invalid amount of bind variables</code> exception.</p>\n\n<p>My query had 5 named parameters, something like this:</p>\n\n<pre><code>private val QUERY = INSERT INTO my_table VALUES (:a, :b, :c, :d, :e)\n</code></pre>\n\n<p>I was trying to execute against it like this:</p>\n\n<pre><code>session.execute(QUERY, mapOf(\"a\" to 1, \"b\" to 2, \"c\" to 3, \"d\" to 4, \"e\" to 5))\n</code></pre>\n\n<p>Since I'm certain that I had the correct number of parameters, and that they were all named correctly (and even in the same order), my best guess is it was instead using the other version of the function, which accepts <code>varargs</code>.</p>\n\n<p>The expected signature of the second parameter is <code>Map&lt;String, Object&gt;</code>, or translated by Kotlin to <code>Map&lt;String!, Object!&gt;!</code>. I tried to cast it to the following:</p>\n\n<ul>\n<li><code>Map&lt;String, Object&gt;</code></li>\n<li><code>Map&lt;String?, Object?&gt;</code></li>\n<li><code>Map&lt;String?, Object?&gt;?</code></li>\n<li><code>Map&lt;String, Any&gt;</code></li>\n<li><code>Map&lt;String?, Any?&gt;</code></li>\n<li><code>Map&lt;String?, Any?&gt;?</code></li>\n</ul>\n\n<p>And none of those seemed to work.</p>\n\n<p>How can I call <code>Session::execute()</code> with a <code>Map</code> as the second parameter?</p>\n"
        },
        {
            "tags": [
                "java",
                "opengl",
                "opengl-es",
                "opengl-es-2.0",
                "lwjgl"
            ],
            "owner": {
                "reputation": 114,
                "user_id": 8365570,
                "user_type": "registered",
                "profile_image": "https://www.gravatar.com/avatar/6e9b4898ee3bdb712d35d5c1d64fa3a9?s=128&d=identicon&r=PG&f=1",
                "display_name": "Санаев",
                "link": "https://stackoverflow.com/users/8365570/%d0%a1%d0%b0%d0%bd%d0%b0%d0%b5%d0%b2"
            },
            "is_answered": true,
            "view_count": 64,
            "answer_count": 2,
            "score": 2,
            "last_activity_date": 1526415876,
            "creation_date": 1526390410,
            "last_edit_date": 1526415876,
            "question_id": 50351326,
            "body_markdown": "I have mouse coordinates: `mousePos`, a matrix view `view`, and a perspective projection matrix `pMatrix`.\r\n\r\nI translate the coordinates into the world: I find the inverse projection matrix and the inverse matrix view and multiply by the coordinates of the mouse. The coordinates of the origin is `z = 4`, the coordinates of the end is `z = -100`.\r\n\r\nIn the first case, I get the coordinates `mouseDir1 = (-0.1985 0.02887 4)`, and in the second case `mouseDir2 = (-0.1985 0.02887 -100)`.\r\n\r\n\r\nWhy are the coordinates `x, y` the same?\r\n\r\n    private Vector3f getCoord(MouseInput mouseInput,float z){\r\n            int wdwWitdh = 640;\r\n            int wdwHeight =640;\r\n    \r\n            Vector2d mousePos = mouseInput.getCurrentPos();\r\n            float x = (float)(2 * mousePos.x) / (float)wdwWitdh - 1.0f;\r\n            float y = 1.0f - (float)(2 * mousePos.y) / (float)wdwHeight;\r\n    \r\n            Matrix4f invProjectionMatrix = new Matrix4f();\r\n            invProjectionMatrix.set(pMatrix);\r\n            invProjectionMatrix.invert();\r\n            Vector4f tmpVec = new Vector4f();\r\n            tmpVec.set(x, y, z, 0);\r\n            tmpVec.mul(invProjectionMatrix);\r\n            tmpVec.z = z;\r\n            tmpVec.w = 0.0f;\r\n    \r\n            Matrix4f viewMatrix = new Matrix4f().set(view);\r\n            Matrix4f invViewMatrix = new Matrix4f();\r\n            invViewMatrix.set(viewMatrix);\r\n            invViewMatrix.invert();\r\n            tmpVec.mul(invViewMatrix);\r\n            Vector3f mouseDir1 = new Vector3f();\r\n            mouseDir1.set(tmpVec.x, tmpVec.y, tmpVec.z);\r\n    \r\n            ///ТЕСТОВАЯ ПРОВЕРКА Z=-100;\r\n            //конеч координаты луча\r\n            Vector4f tmpVec1 = new Vector4f();\r\n            tmpVec1.set(x, y, -100, 1.0f);\r\n            tmpVec1.mul(invProjectionMatrix);\r\n            tmpVec1.z =-100f;\r\n            tmpVec1.w = 0.0f;\r\n            tmpVec1.mul(invViewMatrix);\r\n            Vector3f mouseDir2 = new Vector3f();\r\n            mouseDir2.set(tmpVec1.x, tmpVec1.y, tmpVec1.z);\r\n            System.out.println();\r\n            return mouseDir1;\r\n        }\r\n\r\n**Output ray:**\r\n[![enter image description here][1]][1]\r\n\r\n\r\n\r\n  [1]: https://i.stack.imgur.com/jsEVh.png\r\n",
            "link": "https://stackoverflow.com/questions/50351326/translation-to-world-coordinates",
            "title": "Translation to world coordinates",
            "body": "<p>I have mouse coordinates: <code>mousePos</code>, a matrix view <code>view</code>, and a perspective projection matrix <code>pMatrix</code>.</p>\n\n<p>I translate the coordinates into the world: I find the inverse projection matrix and the inverse matrix view and multiply by the coordinates of the mouse. The coordinates of the origin is <code>z = 4</code>, the coordinates of the end is <code>z = -100</code>.</p>\n\n<p>In the first case, I get the coordinates <code>mouseDir1 = (-0.1985 0.02887 4)</code>, and in the second case <code>mouseDir2 = (-0.1985 0.02887 -100)</code>.</p>\n\n<p>Why are the coordinates <code>x, y</code> the same?</p>\n\n<pre><code>private Vector3f getCoord(MouseInput mouseInput,float z){\n        int wdwWitdh = 640;\n        int wdwHeight =640;\n\n        Vector2d mousePos = mouseInput.getCurrentPos();\n        float x = (float)(2 * mousePos.x) / (float)wdwWitdh - 1.0f;\n        float y = 1.0f - (float)(2 * mousePos.y) / (float)wdwHeight;\n\n        Matrix4f invProjectionMatrix = new Matrix4f();\n        invProjectionMatrix.set(pMatrix);\n        invProjectionMatrix.invert();\n        Vector4f tmpVec = new Vector4f();\n        tmpVec.set(x, y, z, 0);\n        tmpVec.mul(invProjectionMatrix);\n        tmpVec.z = z;\n        tmpVec.w = 0.0f;\n\n        Matrix4f viewMatrix = new Matrix4f().set(view);\n        Matrix4f invViewMatrix = new Matrix4f();\n        invViewMatrix.set(viewMatrix);\n        invViewMatrix.invert();\n        tmpVec.mul(invViewMatrix);\n        Vector3f mouseDir1 = new Vector3f();\n        mouseDir1.set(tmpVec.x, tmpVec.y, tmpVec.z);\n\n        ///ТЕСТОВАЯ ПРОВЕРКА Z=-100;\n        //конеч координаты луча\n        Vector4f tmpVec1 = new Vector4f();\n        tmpVec1.set(x, y, -100, 1.0f);\n        tmpVec1.mul(invProjectionMatrix);\n        tmpVec1.z =-100f;\n        tmpVec1.w = 0.0f;\n        tmpVec1.mul(invViewMatrix);\n        Vector3f mouseDir2 = new Vector3f();\n        mouseDir2.set(tmpVec1.x, tmpVec1.y, tmpVec1.z);\n        System.out.println();\n        return mouseDir1;\n    }\n</code></pre>\n\n<p><strong>Output ray:</strong>\n<a href=\"https://i.stack.imgur.com/jsEVh.png\" rel=\"nofollow noreferrer\"><img src=\"https://i.stack.imgur.com/jsEVh.png\" alt=\"enter image description here\"></a></p>\n"
        },
        {
            "tags": [
                "ios",
                "objective-c",
                "uiimage"
            ],
            "owner": {
                "reputation": 310,
                "user_id": 6631314,
                "user_type": "registered",
                "accept_rate": 63,
                "profile_image": "https://www.gravatar.com/avatar/68a9aec1aca837f507b01299bd81ad0b?s=128&d=identicon&r=PG&f=1",
                "display_name": "user6631314",
                "link": "https://stackoverflow.com/users/6631314/user6631314"
            },
            "is_answered": false,
            "view_count": 34,
            "answer_count": 1,
            "score": 0,
            "last_activity_date": 1526415841,
            "creation_date": 1526339717,
            "last_edit_date": 1526340713,
            "question_id": 50340278,
            "body_markdown": "I have created a tableview programmatically. I set the size for an image using the following code and then place various images in it. Most images display fine but occasionally one will display too narrow or way too wide.  \r\n\r\nSimilar code works fine for tableviews I have created in storyboard except that in that case I set the image size in storyboard, whereas here, I am setting it as shown below.  Here is my code:\r\n\r\n    //make tableview\r\n    \r\n    self.myTableView = [[UITableView alloc] initWithFrame:\r\n                                  CGRectMake(160, 114, 140, 100) style:UITableViewStylePlain];\r\n        _myTableView.separatorStyle = UITableViewCellSeparatorStyleNone;\r\n           _myTableView.rowHeight=24;\r\n       \r\n    //In cellforrowatindexpath\r\n      cell.imageView.frame = CGRectMake(0,0,24,24);\r\n      float imageWidth =cell.imageView.frame.size.width;\r\n    //THis is always 24 due to framesize set above but I have it here to check\r\n     cell.imageView.layer.cornerRadius = cell.imageView.frame.size.width / 2;\r\n     cell.imageView.contentMode = UIViewContentModeScaleAspectFill;\r\n     cell.imageView.clipsToBounds = YES;\r\n     [cell.imageView layoutIfNeeded];//THIS HAS NO EFFECT\r\n      \r\nCan anyone suggest what might be going on here?  Thanks in advance for any suggestions.\r\n\r\n[![enter image description here][1]][1]\r\n\r\n\r\n  [1]: https://i.stack.imgur.com/bSEGg.png",
            "link": "https://stackoverflow.com/questions/50340278/images-appearing-in-different-widths-in-cell-of-programatic-tableview",
            "title": "Images Appearing in Different Widths in Cell of Programatic TableView",
            "body": "<p>I have created a tableview programmatically. I set the size for an image using the following code and then place various images in it. Most images display fine but occasionally one will display too narrow or way too wide.  </p>\n\n<p>Similar code works fine for tableviews I have created in storyboard except that in that case I set the image size in storyboard, whereas here, I am setting it as shown below.  Here is my code:</p>\n\n<pre><code>//make tableview\n\nself.myTableView = [[UITableView alloc] initWithFrame:\n                              CGRectMake(160, 114, 140, 100) style:UITableViewStylePlain];\n    _myTableView.separatorStyle = UITableViewCellSeparatorStyleNone;\n       _myTableView.rowHeight=24;\n\n//In cellforrowatindexpath\n  cell.imageView.frame = CGRectMake(0,0,24,24);\n  float imageWidth =cell.imageView.frame.size.width;\n//THis is always 24 due to framesize set above but I have it here to check\n cell.imageView.layer.cornerRadius = cell.imageView.frame.size.width / 2;\n cell.imageView.contentMode = UIViewContentModeScaleAspectFill;\n cell.imageView.clipsToBounds = YES;\n [cell.imageView layoutIfNeeded];//THIS HAS NO EFFECT\n</code></pre>\n\n<p>Can anyone suggest what might be going on here?  Thanks in advance for any suggestions.</p>\n\n<p><a href=\"https://i.stack.imgur.com/bSEGg.png\" rel=\"nofollow noreferrer\"><img src=\"https://i.stack.imgur.com/bSEGg.png\" alt=\"enter image description here\"></a></p>\n"
        },
        {
            "tags": [
                "python-2.7",
                "can"
            ],
            "owner": {
                "reputation": 8,
                "user_id": 9796330,
                "user_type": "registered",
                "profile_image": "https://lh3.googleusercontent.com/-XdUIqdMkCWA/AAAAAAAAAAI/AAAAAAAAAAA/4252rscbv5M/photo.jpg?sz=128",
                "display_name": "Jeremy Xu",
                "link": "https://stackoverflow.com/users/9796330/jeremy-xu"
            },
            "is_answered": true,
            "view_count": 12,
            "accepted_answer_id": 50358293,
            "answer_count": 1,
            "score": 1,
            "last_activity_date": 1526415839,
            "creation_date": 1526413777,
            "question_id": 50358161,
            "body_markdown": "I am writing a simple python script to read data from a CAN bus. I used the python_can library for python 2.7. It contains a class called CANopenNode and inside the __init_ method I write:\r\n\r\n    self.bus = can.interface.Bus(channel = &#39;can0&#39;, bustype = &#39;socketcan_native&#39;)\r\nwhen I run the code it reports the error like this:\r\n\r\n    File &quot;/home/jxu/.local/lib/python2.7/site-packages/can/interface.py&quot;, line 87, in __new__\r\n    return cls(channel, **kwargs)\r\n    File &quot;/home/jxu/.local/lib/python2.7/site-packages/can/interfaces/socketcan/socketcan_native.py&quot;, line 416, in __init__\r\n    self.socket = create_socket(CAN_RAW)\r\n    File &quot;/home/jxu/.local/lib/python2.7/site-packages/can/interfaces/socketcan/socketcan_native.py&quot;, line 305, in create_socket\r\n    if can_protocol is None or can_protocol == socket.CAN_RAW:\r\n    AttributeError: &#39;module&#39; object has no attribute &#39;CAN_RAW&#39;\r\nThe relative lines of code in create_socket (inside library) looks like this:\r\n\r\n    if can_protocol is None or can_protocol == socket.CAN_RAW:\r\n        can_protocol = socket.CAN_RAW\r\n        socket_type = socket.SOCK_RAW\r\n    elif can_protocol == socket.CAN_BCM:\r\n        can_protocol = socket.CAN_BCM\r\n        socket_type = socket.SOCK_DGRAM\r\n    sock = socket.socket(socket.PF_CAN, socket_type, can_protocol)\r\n\r\nThe library files seem no wrong:\r\n\r\n    &gt;&gt;&gt; import socket\r\n    &gt;&gt;&gt; print socket.__file__\r\n    /usr/lib/python2.7/socket.pyc\r\nCan anyone explain what the CAN_RAW means here and why this error is, thanks!\r\n",
            "link": "https://stackoverflow.com/questions/50358161/attributeerror-module-object-has-no-attribute-can-raw",
            "title": "AttributeError: &#39;module&#39; object has no attribute &#39;CAN_RAW&#39;",
            "body": "<p>I am writing a simple python script to read data from a CAN bus. I used the python_can library for python 2.7. It contains a class called CANopenNode and inside the __init_ method I write:</p>\n\n<pre><code>self.bus = can.interface.Bus(channel = 'can0', bustype = 'socketcan_native')\n</code></pre>\n\n<p>when I run the code it reports the error like this:</p>\n\n<pre><code>File \"/home/jxu/.local/lib/python2.7/site-packages/can/interface.py\", line 87, in __new__\nreturn cls(channel, **kwargs)\nFile \"/home/jxu/.local/lib/python2.7/site-packages/can/interfaces/socketcan/socketcan_native.py\", line 416, in __init__\nself.socket = create_socket(CAN_RAW)\nFile \"/home/jxu/.local/lib/python2.7/site-packages/can/interfaces/socketcan/socketcan_native.py\", line 305, in create_socket\nif can_protocol is None or can_protocol == socket.CAN_RAW:\nAttributeError: 'module' object has no attribute 'CAN_RAW'\n</code></pre>\n\n<p>The relative lines of code in create_socket (inside library) looks like this:</p>\n\n<pre><code>if can_protocol is None or can_protocol == socket.CAN_RAW:\n    can_protocol = socket.CAN_RAW\n    socket_type = socket.SOCK_RAW\nelif can_protocol == socket.CAN_BCM:\n    can_protocol = socket.CAN_BCM\n    socket_type = socket.SOCK_DGRAM\nsock = socket.socket(socket.PF_CAN, socket_type, can_protocol)\n</code></pre>\n\n<p>The library files seem no wrong:</p>\n\n<pre><code>&gt;&gt;&gt; import socket\n&gt;&gt;&gt; print socket.__file__\n/usr/lib/python2.7/socket.pyc\n</code></pre>\n\n<p>Can anyone explain what the CAN_RAW means here and why this error is, thanks!</p>\n"
        },
        {
            "tags": [
                "android",
                "reactivex"
            ],
            "owner": {
                "reputation": 137,
                "user_id": 7564126,
                "user_type": "registered",
                "accept_rate": 57,
                "profile_image": "https://www.gravatar.com/avatar/971ce542fb730b6e4b06d66e6b6240ed?s=128&d=identicon&r=PG&f=1",
                "display_name": "Tarcisio Wensing",
                "link": "https://stackoverflow.com/users/7564126/tarcisio-wensing"
            },
            "is_answered": false,
            "view_count": 18,
            "answer_count": 1,
            "score": 0,
            "last_activity_date": 1526415838,
            "creation_date": 1526073938,
            "question_id": 50300471,
            "body_markdown": "i must do some retrofit calls from a list of ids (can be up to 50 easy)\r\n\r\nmy actual code:\r\n\r\n    idsList\r\n    .subscribeOn(scheduler.computation())\r\n                    .observeOn(scheduler.ui())\r\n                    .flatMapIterable { ids -&gt; ids }\r\n                    .flatMap { id -&gt; requestEmployeeData(id) }\r\n                    .subscribe(\r\n                            {\r\n                                success -&gt;\r\n                                println(&quot;success&quot;)\r\n                            },\r\n                            { error -&gt;\r\n                                error.printStackTrace()\r\n                            }\r\n    )\r\n\r\n\r\n\r\nthe issue with this is that it spawn all the requests in the same time and i dont want this behaviour, i need it to do the first one, wait response, process, finish, do the next one till the end\r\n\r\n\r\nis there any way to only go to the next item on idsList only when the previously ```:Observable&lt;Response&lt;SyncEmployeeData&gt;&gt;``` ends?",
            "link": "https://stackoverflow.com/questions/50300471/rxjava-do-multiple-requests-in-serialized-way",
            "title": "RxJava do multiple requests in serialized way",
            "body": "<p>i must do some retrofit calls from a list of ids (can be up to 50 easy)</p>\n\n<p>my actual code:</p>\n\n<pre><code>idsList\n.subscribeOn(scheduler.computation())\n                .observeOn(scheduler.ui())\n                .flatMapIterable { ids -&gt; ids }\n                .flatMap { id -&gt; requestEmployeeData(id) }\n                .subscribe(\n                        {\n                            success -&gt;\n                            println(\"success\")\n                        },\n                        { error -&gt;\n                            error.printStackTrace()\n                        }\n)\n</code></pre>\n\n<p>the issue with this is that it spawn all the requests in the same time and i dont want this behaviour, i need it to do the first one, wait response, process, finish, do the next one till the end</p>\n\n<p>is there any way to only go to the next item on idsList only when the previously <code>:Observable&lt;Response&lt;SyncEmployeeData&gt;&gt;</code> ends?</p>\n"
        },
        {
            "tags": [
                "c#",
                "azure",
                "visual-studio-2017",
                "azure-functions",
                "solution"
            ],
            "owner": {
                "reputation": 1,
                "user_id": 5758375,
                "user_type": "registered",
                "profile_image": "https://www.gravatar.com/avatar/46deb18a2fdab23f7ae95a8d93289089?s=128&d=identicon&r=PG&f=1",
                "display_name": "norm",
                "link": "https://stackoverflow.com/users/5758375/norm"
            },
            "is_answered": false,
            "view_count": 29,
            "answer_count": 1,
            "score": -1,
            "last_activity_date": 1526415833,
            "creation_date": 1526407424,
            "last_edit_date": 1526408534,
            "question_id": 50356646,
            "body_markdown": "The function project (.csproj) where created in a solution. I deleted the solution and tried to load the function project in another solution.\r\n\r\nResult: When I press F5 to debug, Visual Studio do not recognize it as an Azure Function.\r\n\r\n[![Screenshot][1]][1]\r\n\r\n\r\n  [1]: https://i.stack.imgur.com/YhQPO.png\r\n\r\n\r\nHow can I tell Visual Studio this project is an Azure Function and not a regular class librairy?\r\n\r\n",
            "link": "https://stackoverflow.com/questions/50356646/cannot-debug-azure-function-with-visual-studio-2017",
            "title": "Cannot debug Azure Function with Visual Studio 2017",
            "body": "<p>The function project (.csproj) where created in a solution. I deleted the solution and tried to load the function project in another solution.</p>\n\n<p>Result: When I press F5 to debug, Visual Studio do not recognize it as an Azure Function.</p>\n\n<p><a href=\"https://i.stack.imgur.com/YhQPO.png\" rel=\"nofollow noreferrer\"><img src=\"https://i.stack.imgur.com/YhQPO.png\" alt=\"Screenshot\"></a></p>\n\n<p>How can I tell Visual Studio this project is an Azure Function and not a regular class librairy?</p>\n"
        },
        {
            "tags": [
                "c#",
                "file",
                "office-interop",
                "arabic",
                "arabic-support"
            ],
            "owner": {
                "reputation": 397,
                "user_id": 1719240,
                "user_type": "registered",
                "accept_rate": 60,
                "profile_image": "https://i.stack.imgur.com/qoKJB.jpg?s=128&g=1",
                "display_name": "madan",
                "link": "https://stackoverflow.com/users/1719240/madan"
            },
            "is_answered": false,
            "view_count": 45,
            "answer_count": 1,
            "score": 0,
            "last_activity_date": 1526415832,
            "creation_date": 1526391573,
            "last_edit_date": 1526411570,
            "question_id": 50351763,
            "body_markdown": "In my current work I need to read content from a `.docx` file and write its content to a plain Text file, which will be input for other task.&lt;br&gt;\r\n\r\nBelow is my sample code which work for me for most of language but for arabic I got an issue as arabic is right to left.&lt;br&gt;\r\n\r\n    StringBuilder docText = new StringBuilder();\r\n    Document currentDoc = word.Documents.Open(file.FullName);\r\n    foreach (Paragraph p in currentDoc.Paragraphs)\r\n    {\r\n    \tstring paraText = p.Range.Text;\r\n    \tdocText.AppendLine(paraText);\r\n    }\r\n    currentDoc.Close(false);\r\n    if (docText.Length &gt; 0)\r\n    {\r\n    \tstring outputTxtpath = output + @&quot;\\&quot; + Path.GetFileNameWithoutExtension(file.Name) + &quot;.txt&quot;;\r\n    \tFile.WriteAllText(outputTxtpath, docText.ToString(), Encoding.UTF8);\r\n    }\r\n\r\n\r\nFor arabic I am facing 2 issues: &lt;br/&gt;\r\n\r\n 1. Text in txt file is in left indentation.&lt;br&gt;\r\n 2. If a paragraph is starting with a number (Image 1), then that number come left side in the text file (Image 2), whereas other arabic character sequence is as it is, which is a issue for my 2nd tool.&lt;br&gt;\r\n\r\nSample input (`.docx` file) and output (`.txt` file) uploaded to below location. searched a lot for the issue but everywhere I am getting a solution for how to read a Arabic text file.&lt;br&gt;\r\n\r\n\r\nDocx File : https://1drv.ms/w/s!Ah-Jh2Ok5SuHgQEl90D9NXlM0CJw&lt;br/&gt;\r\nText File : https://1drv.ms/t/s!Ah-Jh2Ok5SuHgQAEPZur1A3379mr\r\n\r\n[![OriginalText from DOCX file][1]][1]&lt;br&gt;\r\n[![Converted Text from TXT file][2]][2]&lt;br&gt;\r\n\r\n\r\n  [1]: https://i.stack.imgur.com/cu66v.png\r\n  [2]: https://i.stack.imgur.com/CFef5.png",
            "link": "https://stackoverflow.com/questions/50351763/how-to-write-a-arabic-text-file-with-indentation",
            "title": "How to write a Arabic text file with indentation?",
            "body": "<p>In my current work I need to read content from a <code>.docx</code> file and write its content to a plain Text file, which will be input for other task.<br></p>\n\n<p>Below is my sample code which work for me for most of language but for arabic I got an issue as arabic is right to left.<br></p>\n\n<pre><code>StringBuilder docText = new StringBuilder();\nDocument currentDoc = word.Documents.Open(file.FullName);\nforeach (Paragraph p in currentDoc.Paragraphs)\n{\n    string paraText = p.Range.Text;\n    docText.AppendLine(paraText);\n}\ncurrentDoc.Close(false);\nif (docText.Length &gt; 0)\n{\n    string outputTxtpath = output + @\"\\\" + Path.GetFileNameWithoutExtension(file.Name) + \".txt\";\n    File.WriteAllText(outputTxtpath, docText.ToString(), Encoding.UTF8);\n}\n</code></pre>\n\n<p>For arabic I am facing 2 issues: <br/></p>\n\n<ol>\n<li>Text in txt file is in left indentation.<br></li>\n<li>If a paragraph is starting with a number (Image 1), then that number come left side in the text file (Image 2), whereas other arabic character sequence is as it is, which is a issue for my 2nd tool.<br></li>\n</ol>\n\n<p>Sample input (<code>.docx</code> file) and output (<code>.txt</code> file) uploaded to below location. searched a lot for the issue but everywhere I am getting a solution for how to read a Arabic text file.<br></p>\n\n<p>Docx File : <a href=\"https://1drv.ms/w/s!Ah-Jh2Ok5SuHgQEl90D9NXlM0CJw\" rel=\"nofollow noreferrer\">https://1drv.ms/w/s!Ah-Jh2Ok5SuHgQEl90D9NXlM0CJw</a><br/>\nText File : <a href=\"https://1drv.ms/t/s!Ah-Jh2Ok5SuHgQAEPZur1A3379mr\" rel=\"nofollow noreferrer\">https://1drv.ms/t/s!Ah-Jh2Ok5SuHgQAEPZur1A3379mr</a></p>\n\n<p><a href=\"https://i.stack.imgur.com/cu66v.png\" rel=\"nofollow noreferrer\"><img src=\"https://i.stack.imgur.com/cu66v.png\" alt=\"OriginalText from DOCX file\"></a><br>\n<a href=\"https://i.stack.imgur.com/CFef5.png\" rel=\"nofollow noreferrer\"><img src=\"https://i.stack.imgur.com/CFef5.png\" alt=\"Converted Text from TXT file\"></a><br></p>\n"
        },
        {
            "tags": [
                "python",
                "matplotlib",
                "visualization"
            ],
            "owner": {
                "reputation": 51,
                "user_id": 8879643,
                "user_type": "registered",
                "accept_rate": 100,
                "profile_image": "https://www.gravatar.com/avatar/89ef3c2e7f94c2b504d348f14c3fa453?s=128&d=identicon&r=PG&f=1",
                "display_name": "AToe",
                "link": "https://stackoverflow.com/users/8879643/atoe"
            },
            "is_answered": true,
            "view_count": 48,
            "accepted_answer_id": 50358336,
            "answer_count": 1,
            "score": 4,
            "last_activity_date": 1526415824,
            "creation_date": 1526412491,
            "question_id": 50357852,
            "body_markdown": "I&#39;m creating a gauge using matplotlib pie as the foundation:\r\n\r\n    import matplotlib.pyplot as plt\r\n    import math    \r\n    \r\n    theta = 0.2\r\n        \r\n    group_size=[10,10,10,10,10,50]\r\n        \r\n    mid = [18,54,90,126,162]\r\n    \r\n    from textwrap import wrap\r\n    labels=[&#39;1&#39;,&#39;2&#39;,&#39;3&#39;,&#39;4&#39;,&#39;5&#39;,&#39;&#39;]\r\n    labels = [ &#39;\\n&#39;.join(wrap(l, 9)) for l in labels ]\r\n    \r\n    fig, ax = plt.subplots()\r\n    \r\n    ax.axis(&#39;equal&#39;)\r\n    pie3 = ax.pie(group_size, radius=2.2, colors=[&#39;k&#39;] ,startangle=180,counterclock=False)\r\n    \r\n    my_circle=plt.Circle( (0,0), 0.4, color=&#39;white&#39;)\r\n    p=plt.gcf()\r\n    p.gca().add_artist(my_circle)\r\n    \r\n    pie4 = ax.pie([10,10,10,10,10,50], radius=2,  labeldistance=0.9, labels=labels,\r\n                  startangle=180,rotatelabels =True,counterclock=False)\r\n    \r\n    plt.setp(pie4[1], rotation_mode=&quot;anchor&quot;, ha=&quot;center&quot;, va=&quot;center&quot;)\r\n    for tx in pie4[1]:\r\n        font = tx.get_fontsize()\r\n        tx.set_fontsize(12)\r\n        rot = tx.get_rotation()\r\n        tx.set_rotation(rot+90+(1-rot//180)*180)\r\n    \r\n    for pie_wedge in pie3[0]:\r\n        pie_wedge.set_edgecolor(&#39;white&#39;)\r\n    \r\n    for pie_wedge in pie4[0]:\r\n        pie_wedge.set_edgecolor(&#39;white&#39;)\r\n        pie_wedge.set_linewidth(4)\r\n    \r\n    #arrow bottom circle\r\n    ax.add_patch(Circle((0, 0), radius=0.15, facecolor=&#39;k&#39;,zorder=12))\r\n    ax.add_patch(Circle((0, 0), radius=0.13, facecolor=&#39;w&#39;,zorder=13))\r\n    \r\n    ax.arrow(0, 0, 1.7 * np.sin(math.radians(theta*180-90)), 1.7 * np.cos(math.radians(theta*180-90)), \\\r\n                     width=0.15, head_width=0.15, head_length=0.2, fc=&#39;k&#39;, ec=&#39;k&#39;)\r\n    \r\n    theta2 = 0.45\r\n    \r\n    ax.arrow(0, 0, 1.7 * np.sin(math.radians(theta2*180-90)), 1.7 * np.cos(math.radians(theta2*180-90)), \\\r\n                     width=0.15, head_width=0.15, head_length=0.2, fc=&#39;k&#39;, ec=&#39;k&#39;,zorder=20)\r\n\r\nNow the problem is my gauge arrow (in red) seems to get cut off at the top if it is sitting between 2 and 4. Almost as if there is an imaginary line between the left corner of the 2 wedge and the right corner of the 4 wedge. As you can see with the black arrow, no issue there. \r\n\r\n[![enter image description here][1]][1]\r\n\r\nDoes anyone know what is happening here? I suspect a possible figsize issue?\r\n\r\n\r\n  [1]: https://i.stack.imgur.com/ap0fN.png",
            "link": "https://stackoverflow.com/questions/50357852/strange-pie-graph-cut-off",
            "title": "Strange pie graph cut off",
            "body": "<p>I'm creating a gauge using matplotlib pie as the foundation:</p>\n\n<pre><code>import matplotlib.pyplot as plt\nimport math    \n\ntheta = 0.2\n\ngroup_size=[10,10,10,10,10,50]\n\nmid = [18,54,90,126,162]\n\nfrom textwrap import wrap\nlabels=['1','2','3','4','5','']\nlabels = [ '\\n'.join(wrap(l, 9)) for l in labels ]\n\nfig, ax = plt.subplots()\n\nax.axis('equal')\npie3 = ax.pie(group_size, radius=2.2, colors=['k'] ,startangle=180,counterclock=False)\n\nmy_circle=plt.Circle( (0,0), 0.4, color='white')\np=plt.gcf()\np.gca().add_artist(my_circle)\n\npie4 = ax.pie([10,10,10,10,10,50], radius=2,  labeldistance=0.9, labels=labels,\n              startangle=180,rotatelabels =True,counterclock=False)\n\nplt.setp(pie4[1], rotation_mode=\"anchor\", ha=\"center\", va=\"center\")\nfor tx in pie4[1]:\n    font = tx.get_fontsize()\n    tx.set_fontsize(12)\n    rot = tx.get_rotation()\n    tx.set_rotation(rot+90+(1-rot//180)*180)\n\nfor pie_wedge in pie3[0]:\n    pie_wedge.set_edgecolor('white')\n\nfor pie_wedge in pie4[0]:\n    pie_wedge.set_edgecolor('white')\n    pie_wedge.set_linewidth(4)\n\n#arrow bottom circle\nax.add_patch(Circle((0, 0), radius=0.15, facecolor='k',zorder=12))\nax.add_patch(Circle((0, 0), radius=0.13, facecolor='w',zorder=13))\n\nax.arrow(0, 0, 1.7 * np.sin(math.radians(theta*180-90)), 1.7 * np.cos(math.radians(theta*180-90)), \\\n                 width=0.15, head_width=0.15, head_length=0.2, fc='k', ec='k')\n\ntheta2 = 0.45\n\nax.arrow(0, 0, 1.7 * np.sin(math.radians(theta2*180-90)), 1.7 * np.cos(math.radians(theta2*180-90)), \\\n                 width=0.15, head_width=0.15, head_length=0.2, fc='k', ec='k',zorder=20)\n</code></pre>\n\n<p>Now the problem is my gauge arrow (in red) seems to get cut off at the top if it is sitting between 2 and 4. Almost as if there is an imaginary line between the left corner of the 2 wedge and the right corner of the 4 wedge. As you can see with the black arrow, no issue there. </p>\n\n<p><a href=\"https://i.stack.imgur.com/ap0fN.png\" rel=\"nofollow noreferrer\"><img src=\"https://i.stack.imgur.com/ap0fN.png\" alt=\"enter image description here\"></a></p>\n\n<p>Does anyone know what is happening here? I suspect a possible figsize issue?</p>\n"
        },
        {
            "tags": [
                "sql-server",
                "triggers",
                "primary-key"
            ],
            "owner": {
                "reputation": 101,
                "user_id": 9795061,
                "user_type": "registered",
                "profile_image": "https://i.stack.imgur.com/ubcVp.jpg?s=128&g=1",
                "display_name": "Ahmad saleh",
                "link": "https://stackoverflow.com/users/9795061/ahmad-saleh"
            },
            "is_answered": false,
            "view_count": 14,
            "answer_count": 0,
            "score": 0,
            "last_activity_date": 1526415820,
            "creation_date": 1526399253,
            "last_edit_date": 1526415820,
            "question_id": 50354375,
            "body_markdown": "I have the following 2 tables:\r\n\r\n1. `Order` with the following columns: `OrderID` as PK, `Order Name`, etc.\r\n\r\n2. `Delivery` with the following columns: `DeliveryID` as PK, `OrderID`, `date`, etc.\r\n\r\nI wrote after update trigger on the `Order` table that inserts a new row to the `Delivery` table and fills it with the `OrderID`:\r\n\r\n    create Trigger CopyOrderID_To_DeliveryTable\r\n    on Order\r\n    for insert \r\n    as \r\n    begin\r\n        declare @ID int\r\n\r\n        select @ID = REQUESTID from inserted  \r\n    \r\n        insert into Delivery (OrderID) \r\n        values (@ID)\r\n    end\r\n\r\nThis should work fine but the issue here is that the `DeliveryID` column in the `Delivery` table is a non-nullable PK. So whenever I try to update the `Order` table, I get an error message that `DeliveryID` in the `Delivery` table doesn’t allow nulls!\r\n\r\nHow to get rid of this message without altering the `DeliveryID` column?\r\n\r\nA screenshot shows the `DeliveryID` column properties :\r\n\r\n[![enter image description here][1]][1]\r\n\r\n\r\n  [1]: https://i.stack.imgur.com/rBRAb.jpg",
            "link": "https://stackoverflow.com/questions/50354375/after-insert-trigger-on-a-table-that-has-a-primary-key",
            "title": "After insert trigger on a table that has a Primary Key",
            "body": "<p>I have the following 2 tables:</p>\n\n<ol>\n<li><p><code>Order</code> with the following columns: <code>OrderID</code> as PK, <code>Order Name</code>, etc.</p></li>\n<li><p><code>Delivery</code> with the following columns: <code>DeliveryID</code> as PK, <code>OrderID</code>, <code>date</code>, etc.</p></li>\n</ol>\n\n<p>I wrote after update trigger on the <code>Order</code> table that inserts a new row to the <code>Delivery</code> table and fills it with the <code>OrderID</code>:</p>\n\n<pre><code>create Trigger CopyOrderID_To_DeliveryTable\non Order\nfor insert \nas \nbegin\n    declare @ID int\n\n    select @ID = REQUESTID from inserted  \n\n    insert into Delivery (OrderID) \n    values (@ID)\nend\n</code></pre>\n\n<p>This should work fine but the issue here is that the <code>DeliveryID</code> column in the <code>Delivery</code> table is a non-nullable PK. So whenever I try to update the <code>Order</code> table, I get an error message that <code>DeliveryID</code> in the <code>Delivery</code> table doesn’t allow nulls!</p>\n\n<p>How to get rid of this message without altering the <code>DeliveryID</code> column?</p>\n\n<p>A screenshot shows the <code>DeliveryID</code> column properties :</p>\n\n<p><a href=\"https://i.stack.imgur.com/rBRAb.jpg\" rel=\"nofollow noreferrer\"><img src=\"https://i.stack.imgur.com/rBRAb.jpg\" alt=\"enter image description here\"></a></p>\n"
        },
        {
            "tags": [
                "angular",
                "lint"
            ],
            "owner": {
                "reputation": 70,
                "user_id": 518270,
                "user_type": "registered",
                "accept_rate": 57,
                "profile_image": "https://www.gravatar.com/avatar/299de1d017255ddccdc9b3c8933f1fb6?s=128&d=identicon&r=PG",
                "display_name": "mtanzania",
                "link": "https://stackoverflow.com/users/518270/mtanzania"
            },
            "is_answered": false,
            "view_count": 16,
            "answer_count": 0,
            "score": 0,
            "last_activity_date": 1526415817,
            "creation_date": 1526415817,
            "question_id": 50358627,
            "body_markdown": "In our angular 5 project we&#39;ve seen several instances of developers committing html markup in templates with missing closing tags / quotations, etc. Is there a tool to check the templates similar to what lint does for js files?",
            "link": "https://stackoverflow.com/questions/50358627/what-is-best-way-to-validate-angular-template-markup",
            "title": "What is best way to validate angular template markup",
            "body": "<p>In our angular 5 project we've seen several instances of developers committing html markup in templates with missing closing tags / quotations, etc. Is there a tool to check the templates similar to what lint does for js files?</p>\n"
        },
        {
            "tags": [
                "matlab",
                "ms-access",
                "database-connection"
            ],
            "owner": {
                "reputation": 526,
                "user_id": 3230708,
                "user_type": "registered",
                "accept_rate": 71,
                "profile_image": "https://www.gravatar.com/avatar/3ba0a3fe017d7c55f7741bcb315f4824?s=128&d=identicon&r=PG&f=1",
                "display_name": "user36800",
                "link": "https://stackoverflow.com/users/3230708/user36800"
            },
            "is_answered": true,
            "view_count": 54,
            "accepted_answer_id": 50082099,
            "answer_count": 2,
            "score": 1,
            "last_activity_date": 1526415807,
            "creation_date": 1524898352,
            "last_edit_date": 1524950159,
            "question_id": 50073768,
            "body_markdown": "I have used the following code pattern to access my *.accdb files:\r\n\r\n    accdb_path=&#39;C:\\path\\to\\accdb\\file\\wbe3.accdb&#39;;\r\n    accdb_url= [ &#39;jdbc:odbc:Driver={Microsoft Access Driver (*.mdb, *.accdb)};DSN=&#39;&#39;&#39;&#39;;DBQ=&#39; accdb_path ];\r\n    conn = database(&#39;&#39;,&#39;&#39;,&#39;&#39;,&#39;sun.jdbc.odbc.JdbcOdbcDriver&#39;,accdb_url);\r\n\r\nIf instead I want to *create* a new *.accdb file, how would I do that?  There is much on the web about how to connect, but I haven&#39;t found how to create the *.accdb file itself.\r\n\r\nIn case it matters, I want to be able to execute SQL 92 syntax.  I am using Matlab 2015b.  I do not want to use the Matlab GUI for exploring databases.",
            "link": "https://stackoverflow.com/questions/50073768/matlab-create-new-microsoft-access-database-file-accdb",
            "title": "Matlab create new Microsoft Access Database file *.accdb",
            "body": "<p>I have used the following code pattern to access my *.accdb files:</p>\n\n<pre><code>accdb_path='C:\\path\\to\\accdb\\file\\wbe3.accdb';\naccdb_url= [ 'jdbc:odbc:Driver={Microsoft Access Driver (*.mdb, *.accdb)};DSN='''';DBQ=' accdb_path ];\nconn = database('','','','sun.jdbc.odbc.JdbcOdbcDriver',accdb_url);\n</code></pre>\n\n<p>If instead I want to <em>create</em> a new *.accdb file, how would I do that?  There is much on the web about how to connect, but I haven't found how to create the *.accdb file itself.</p>\n\n<p>In case it matters, I want to be able to execute SQL 92 syntax.  I am using Matlab 2015b.  I do not want to use the Matlab GUI for exploring databases.</p>\n"
        },
        {
            "tags": [
                "python",
                "selenium",
                "selenium-webdriver",
                "xpath",
                "webdriverwait"
            ],
            "owner": {
                "reputation": 15,
                "user_id": 9692024,
                "user_type": "registered",
                "profile_image": "https://www.gravatar.com/avatar/8b06378f1523c1325e872396a93b4460?s=128&d=identicon&r=PG&f=1",
                "display_name": "bot1",
                "link": "https://stackoverflow.com/users/9692024/bot1"
            },
            "is_answered": true,
            "view_count": 51,
            "accepted_answer_id": 50358509,
            "answer_count": 1,
            "score": 1,
            "last_activity_date": 1526415804,
            "creation_date": 1526401696,
            "last_edit_date": 1526415710,
            "question_id": 50355120,
            "body_markdown": "I am trying to use selenium python to auto search on yahoo finance. When I type some words a suggestion will pop out like the same on google suggestion.\r\n\r\nhttps://finance.yahoo.com/\r\n\r\nI found a list element with the xpath should be the suggestions made by yahoo:\r\n\r\n    //*[@id=&quot;search-assist-input&quot;]/div[2]/ul\r\n\r\nIt seems the suggestion content are hidden in this list, but it&#39;s invisible, I mean when I click to unfold it, it just goes away. I don&#39;t know if there&#39;s some sort of &#39;always unfold nodes&#39; in firefox or chrome, but these elements seem very hard to reach.\r\nI tried to get all the children under this element, it shows no element can be found:\r\n\r\n    from chrome_driver.chrome import Chrome\r\n\r\n    driver = Chrome().get_driver()\r\n    driver.get(&#39;https://finance.yahoo.com/&#39;)\r\n    driver.find_elements_by_xpath(&quot;//div[@id=&#39;search-assist-input&#39;]/div/input&quot;)[0].send_keys(&#39;goog&#39;)\r\n    x = driver.find_elements_by_xpath(&quot;//div[@data-reactid=&#39;56&#39;]/ul[@data-reactid=&#39;57&#39;]/*&quot;)\r\n\r\nHow can I reach these auto suggestion from the search box?",
            "link": "https://stackoverflow.com/questions/50355120/how-to-scrape-the-yahoo-finance-search-auto-suggestion-result-with-selenium-pyth",
            "title": "How to scrape the yahoo finance search auto suggestion result with selenium python?",
            "body": "<p>I am trying to use selenium python to auto search on yahoo finance. When I type some words a suggestion will pop out like the same on google suggestion.</p>\n\n<p><a href=\"https://finance.yahoo.com/\" rel=\"nofollow noreferrer\">https://finance.yahoo.com/</a></p>\n\n<p>I found a list element with the xpath should be the suggestions made by yahoo:</p>\n\n<pre><code>//*[@id=\"search-assist-input\"]/div[2]/ul\n</code></pre>\n\n<p>It seems the suggestion content are hidden in this list, but it's invisible, I mean when I click to unfold it, it just goes away. I don't know if there's some sort of 'always unfold nodes' in firefox or chrome, but these elements seem very hard to reach.\nI tried to get all the children under this element, it shows no element can be found:</p>\n\n<pre><code>from chrome_driver.chrome import Chrome\n\ndriver = Chrome().get_driver()\ndriver.get('https://finance.yahoo.com/')\ndriver.find_elements_by_xpath(\"//div[@id='search-assist-input']/div/input\")[0].send_keys('goog')\nx = driver.find_elements_by_xpath(\"//div[@data-reactid='56']/ul[@data-reactid='57']/*\")\n</code></pre>\n\n<p>How can I reach these auto suggestion from the search box?</p>\n"
        },
        {
            "tags": [
                "python",
                "pandas",
                "dataframe",
                "subtract"
            ],
            "owner": {
                "reputation": 32,
                "user_id": 7686543,
                "user_type": "registered",
                "accept_rate": 60,
                "profile_image": "https://www.gravatar.com/avatar/82f6a9fcc52bab17caa23002682dc9dd?s=128&d=identicon&r=PG&f=1",
                "display_name": "Research100",
                "link": "https://stackoverflow.com/users/7686543/research100"
            },
            "is_answered": true,
            "view_count": 28,
            "accepted_answer_id": 50358483,
            "answer_count": 2,
            "score": -1,
            "last_activity_date": 1526415801,
            "creation_date": 1526413713,
            "last_edit_date": 1526415419,
            "question_id": 50358140,
            "body_markdown": "I have a DataFrame (Main) as below. The columns have a group classification as described in the Group Dict. There is a second DataFrame with Group values. I need to subtract from each column in Main the value from the corresponding group in Group Value DataFrame. The resulting table is shown below as well. \r\n(Exp: Main[&quot;AAA&quot;]-Group_Value[&quot;Group1&quot;], etc.)\r\nIs there a matrix form of doing this or will I need to use a for-loop?\r\n\r\nCode:\r\n\r\n\r\n    import random\r\n    df = pd.DataFrame(index=pd.date_range(&quot;1/1/2018&quot;,&quot;1/10/2018&quot;), columns= \r\n    [&quot;AAA&quot;,&quot;BBB&quot;,&quot;CCC&quot;,&quot;DDD&quot;])\r\n    df[&quot;AAA&quot;]=range(100,110)\r\n    df[&quot;BBB&quot;]=range(200,210)\r\n    df[&quot;CCC&quot;]=range(300,310)\r\n    df[&quot;DDD&quot;]=range(400,410)\r\n\r\n    Group_Dict = dict({&quot;AAA&quot;:&quot;Group1&quot;,  &quot;BBB&quot;:&quot;Group2&quot;, &quot;CCC&quot;:&quot;Group1&quot;, &quot;DDD&quot;:&quot;Group2&quot;})\r\n\r\n    group_value = pd.DataFrame(index=pd.date_range(&quot;1/1/2018&quot;,&quot;1/10/2018&quot;), columns=[&quot;Group1&quot;,&quot;Group2&quot;])\r\n    group_value[&quot;Group1&quot;]=range(10,29)[::2]\r\n    group_value[&quot;Group2&quot;]=range(100,600)[::50]\r\n\r\n    ## I need to do the following AAA-Group1, BBB-Group2 , CCC-Group1, DDD-Group2 \r\n\r\n\r\n\r\nThank you.\r\n\r\n\r\n&#39;[![enter image description here][1]][1]\r\n\r\n\r\n  [1]: https://i.stack.imgur.com/9OaEP.png",
            "link": "https://stackoverflow.com/questions/50358140/pandas-dataframe-subtract-a-series-using-groupby-classification",
            "title": "Pandas DataFrame subtract a series using groupby classification",
            "body": "<p>I have a DataFrame (Main) as below. The columns have a group classification as described in the Group Dict. There is a second DataFrame with Group values. I need to subtract from each column in Main the value from the corresponding group in Group Value DataFrame. The resulting table is shown below as well. \n(Exp: Main[\"AAA\"]-Group_Value[\"Group1\"], etc.)\nIs there a matrix form of doing this or will I need to use a for-loop?</p>\n\n<p>Code:</p>\n\n<pre><code>import random\ndf = pd.DataFrame(index=pd.date_range(\"1/1/2018\",\"1/10/2018\"), columns= \n[\"AAA\",\"BBB\",\"CCC\",\"DDD\"])\ndf[\"AAA\"]=range(100,110)\ndf[\"BBB\"]=range(200,210)\ndf[\"CCC\"]=range(300,310)\ndf[\"DDD\"]=range(400,410)\n\nGroup_Dict = dict({\"AAA\":\"Group1\",  \"BBB\":\"Group2\", \"CCC\":\"Group1\", \"DDD\":\"Group2\"})\n\ngroup_value = pd.DataFrame(index=pd.date_range(\"1/1/2018\",\"1/10/2018\"), columns=[\"Group1\",\"Group2\"])\ngroup_value[\"Group1\"]=range(10,29)[::2]\ngroup_value[\"Group2\"]=range(100,600)[::50]\n\n## I need to do the following AAA-Group1, BBB-Group2 , CCC-Group1, DDD-Group2 \n</code></pre>\n\n<p>Thank you.</p>\n\n<p>'<a href=\"https://i.stack.imgur.com/9OaEP.png\" rel=\"nofollow noreferrer\"><img src=\"https://i.stack.imgur.com/9OaEP.png\" alt=\"enter image description here\"></a></p>\n"
        },
        {
            "tags": [
                "linux",
                "firmware"
            ],
            "owner": {
                "reputation": 1,
                "user_id": 9592820,
                "user_type": "registered",
                "profile_image": "https://www.gravatar.com/avatar/63e722bc60001bbcdc34520ca7b2207a?s=128&d=identicon&r=PG&f=1",
                "display_name": "Vladimir",
                "link": "https://stackoverflow.com/users/9592820/vladimir"
            },
            "is_answered": false,
            "view_count": 20,
            "closed_date": 1526552331,
            "answer_count": 0,
            "score": -2,
            "last_activity_date": 1526415786,
            "creation_date": 1526415786,
            "question_id": 50358625,
            "body_markdown": "guys! Do you know what is it? If yes, please, answer to me... I had this problem after I plugged my computer in power socket. Did you have something like that before in your life?\n\n\n  [1]: https://i.stack.imgur.com/MGLI2.jpg",
            "link": "https://stackoverflow.com/questions/50358625/firmware-bug-in-linux-mint",
            "closed_reason": "unclear what you&#39;re asking",
            "title": "Firmware bug in Linux Mint",
            "body": "<p>guys! Do you know what is it? If yes, please, answer to me... I had this problem after I plugged my computer in power socket. Did you have something like that before in your life?</p>\n"
        },
        {
            "tags": [
                "r"
            ],
            "owner": {
                "reputation": 14,
                "user_id": 2797552,
                "user_type": "registered",
                "profile_image": "https://www.gravatar.com/avatar/381ca790d841e478a0997cf923196740?s=128&d=identicon&r=PG&f=1",
                "display_name": "Mark K.",
                "link": "https://stackoverflow.com/users/2797552/mark-k"
            },
            "is_answered": true,
            "view_count": 49,
            "accepted_answer_id": 50357265,
            "answer_count": 1,
            "score": -1,
            "last_activity_date": 1526415782,
            "creation_date": 1526407488,
            "last_edit_date": 1526411452,
            "question_id": 50356659,
            "body_markdown": "I have a list object called results. This list contains 2 list, each of those list also have multiple columns. I would like to convert it to data frame that combine those columns on each list. I know that we can&#39;t combine columns with different length, so is there any way to put NA&#39;s for the extra observations. \r\nhere is a small portion of the list object (results)\r\n\r\n    results\t\t\t\r\n    [[1]]\t\t\t\r\n    \t     gene_name\tx1\t                       x2\r\n    gene34556\t gene1\t           0\t            0\r\n    gene11169\t gene2\t 0.098757012\t            0\r\n    gene11319\t gene3\t           0\t            0\r\n    gene1459\t gene3\t           0\t            0\r\n    gen168232\t gene5             0\t            0\r\n    gene2992\t gene6   -1.93960816\t  0.042291503\r\n    gene305454\t gene7\t           0\t            0\r\n    gene3280\t gene8\t           0\t            0\r\n    \t\t\t\r\n    [[2]]\t\t\t\r\n    \t        gene_name\t       x1\t          x2\r\n    gene34556\tgene1\t            0\t           0\r\n    gene11169\tgene2\t -3.785515694\t           0\r\n    gene11319\tgene3\t            0\t           0\r\n    gene1459\tgene4\t            0\t           0\r\n    gene2992\tgene5\t -2.308363477\t-0.267514619\r\n    \r\n   \r\n\r\n",
            "link": "https://stackoverflow.com/questions/50356659/how-to-convert-list-with-multiple-columns-to-data-frame-in-r",
            "title": "How to convert list with multiple columns to data frame in R",
            "body": "<p>I have a list object called results. This list contains 2 list, each of those list also have multiple columns. I would like to convert it to data frame that combine those columns on each list. I know that we can't combine columns with different length, so is there any way to put NA's for the extra observations. \nhere is a small portion of the list object (results)</p>\n\n<pre><code>results         \n[[1]]           \n         gene_name  x1                         x2\ngene34556    gene1             0                0\ngene11169    gene2   0.098757012                0\ngene11319    gene3             0                0\ngene1459     gene3             0                0\ngen168232    gene5             0                0\ngene2992     gene6   -1.93960816      0.042291503\ngene305454   gene7             0                0\ngene3280     gene8             0                0\n\n[[2]]           \n            gene_name          x1             x2\ngene34556   gene1               0              0\ngene11169   gene2    -3.785515694              0\ngene11319   gene3               0              0\ngene1459    gene4               0              0\ngene2992    gene5    -2.308363477   -0.267514619\n</code></pre>\n"
        },
        {
            "tags": [
                "automation",
                "rft"
            ],
            "owner": {
                "reputation": 1,
                "user_id": 9796532,
                "user_type": "registered",
                "profile_image": "https://www.gravatar.com/avatar/1d3709d3aa726c559c9ff3aa2dffa6d2?s=128&d=identicon&r=PG&f=1",
                "display_name": "LOM1689",
                "link": "https://stackoverflow.com/users/9796532/lom1689"
            },
            "is_answered": false,
            "view_count": 9,
            "answer_count": 0,
            "score": 0,
            "last_activity_date": 1526415744,
            "creation_date": 1526415744,
            "question_id": 50358619,
            "body_markdown": "When trying to find the html properties of textbox login and password, the object search does not recognize them.\r\n\r\nDoes anyone know any solution?",
            "link": "https://stackoverflow.com/questions/50358619/rft-object-finder-does-not-recognize-windows-security-pop-up-windows-10",
            "title": "RFT object finder does not recognize windows security pop up windows 10",
            "body": "<p>When trying to find the html properties of textbox login and password, the object search does not recognize them.</p>\n\n<p>Does anyone know any solution?</p>\n"
        },
        {
            "tags": [
                "mysql",
                "node.js",
                "express",
                "model-view-controller",
                "sequelize.js"
            ],
            "owner": {
                "reputation": 7,
                "user_id": 9795341,
                "user_type": "registered",
                "profile_image": "https://www.gravatar.com/avatar/5a6207483d7bee6014502149356b0bf6?s=128&d=identicon&r=PG&f=1",
                "display_name": "varun",
                "link": "https://stackoverflow.com/users/9795341/varun"
            },
            "is_answered": true,
            "view_count": 25,
            "answer_count": 2,
            "score": 2,
            "last_activity_date": 1526415737,
            "creation_date": 1526401173,
            "question_id": 50354950,
            "body_markdown": "I have an Express application where I use Sequelize to interact with my MySQL database. The models just represent the database tables with corresponding fields,  without containing any additional logic. I want my controllers to not be fat and for that I think I should have a layer in between which contains all of the logic and which uses the model to interact with the database.\r\n\r\nIs this a good practice and if it is what should I call this layer and what exactly should it contain?\r\n\r\nThanks, in advance!",
            "link": "https://stackoverflow.com/questions/50354950/should-i-have-a-layer-in-between-my-controller-and-model-in-node-express-app",
            "title": "Should I have a layer in between my controller and model in Node-Express app?",
            "body": "<p>I have an Express application where I use Sequelize to interact with my MySQL database. The models just represent the database tables with corresponding fields,  without containing any additional logic. I want my controllers to not be fat and for that I think I should have a layer in between which contains all of the logic and which uses the model to interact with the database.</p>\n\n<p>Is this a good practice and if it is what should I call this layer and what exactly should it contain?</p>\n\n<p>Thanks, in advance!</p>\n"
        },
        {
            "tags": [
                "django",
                "forms",
                "postgresql",
                "datetime"
            ],
            "owner": {
                "reputation": 28,
                "user_id": 5660974,
                "user_type": "registered",
                "accept_rate": 40,
                "profile_image": "https://www.gravatar.com/avatar/78e8092d4b15b7189cd2568b4983ddec?s=128&d=identicon&r=PG&f=1",
                "display_name": "James",
                "link": "https://stackoverflow.com/users/5660974/james"
            },
            "is_answered": false,
            "view_count": 13,
            "answer_count": 1,
            "score": 1,
            "last_activity_date": 1526415737,
            "creation_date": 1526215882,
            "question_id": 50316314,
            "body_markdown": "I have a model like this:\r\n\r\n    class MyModel(models.Model):\r\n\t    name = models.CharField(max_length=127)\r\n\t    slot = DateTimeRangeField()\r\n\r\nand the form as:\r\n\r\n    class MyModelForm(forms.ModelForm):\r\n\t    class Meta:\r\n\t\t    model = MyModel\r\n    \t\tfields = [&#39;name&#39;, &#39;slot&#39;,]\r\n\r\nThe form renders with 2 inputs for the lower and upper bounds of the date range. How can I render them as 4 inputs in the template? (lower date, lower time, upper date, upper time). Is there a widget that can do this?",
            "link": "https://stackoverflow.com/questions/50316314/displaying-django-datetimerangefield-in-form-as-separate-inputs",
            "title": "Displaying django DateTimeRangeField() in form as separate inputs",
            "body": "<p>I have a model like this:</p>\n\n<pre><code>class MyModel(models.Model):\n    name = models.CharField(max_length=127)\n    slot = DateTimeRangeField()\n</code></pre>\n\n<p>and the form as:</p>\n\n<pre><code>class MyModelForm(forms.ModelForm):\n    class Meta:\n        model = MyModel\n        fields = ['name', 'slot',]\n</code></pre>\n\n<p>The form renders with 2 inputs for the lower and upper bounds of the date range. How can I render them as 4 inputs in the template? (lower date, lower time, upper date, upper time). Is there a widget that can do this?</p>\n"
        },
        {
            "tags": [
                "amazon-web-services",
                "amazon-sns",
                "amazon-cloudwatch"
            ],
            "owner": {
                "reputation": 14956,
                "user_id": 1308743,
                "user_type": "registered",
                "accept_rate": 69,
                "profile_image": "https://www.gravatar.com/avatar/92ec7a7b317928af4b975419239ffa99?s=128&d=identicon&r=PG&f=1",
                "display_name": "Kyle",
                "link": "https://stackoverflow.com/users/1308743/kyle"
            },
            "is_answered": false,
            "view_count": 7,
            "answer_count": 0,
            "score": 0,
            "last_activity_date": 1526415735,
            "creation_date": 1526415735,
            "question_id": 50358615,
            "body_markdown": "I am trying to create CloudWatch alarms in N. California region because that is where my EC2s are.\r\n\r\nI want to send a notification to an SNS topic that supports both email and SMS, which n. california does not, so I created an SNS topic in Oregon instead which supports both. \r\n\r\nThe topic does not appear in the &quot;Send Notification To:&quot; dropdown, reading online it seems like i&#39;m supposed to be able to click &#39;enter list&#39; and put in an ARN from another region, however when I do this I get an error saying:\r\n\r\n\r\n    Error:\r\n    There was an error saving the alarm. Please try again.",
            "link": "https://stackoverflow.com/questions/50358615/is-it-possible-to-have-aws-cloudwatch-alarm-send-a-notification-to-an-sns-topic",
            "title": "Is it possible to have AWS CloudWatch Alarm send a notification to an SNS topic in another region?",
            "body": "<p>I am trying to create CloudWatch alarms in N. California region because that is where my EC2s are.</p>\n\n<p>I want to send a notification to an SNS topic that supports both email and SMS, which n. california does not, so I created an SNS topic in Oregon instead which supports both. </p>\n\n<p>The topic does not appear in the \"Send Notification To:\" dropdown, reading online it seems like i'm supposed to be able to click 'enter list' and put in an ARN from another region, however when I do this I get an error saying:</p>\n\n<pre><code>Error:\nThere was an error saving the alarm. Please try again.\n</code></pre>\n"
        },
        {
            "tags": [
                "c#",
                "uwp",
                "windows-10-universal",
                "uwp-community-toolkit"
            ],
            "owner": {
                "reputation": 1039,
                "user_id": 1869918,
                "user_type": "registered",
                "accept_rate": 29,
                "profile_image": "https://www.gravatar.com/avatar/3f50a2bdd4c896373ccdc93eda3dcc89?s=128&d=identicon&r=PG&f=1",
                "display_name": "Ankit Dhadse",
                "link": "https://stackoverflow.com/users/1869918/ankit-dhadse"
            },
            "is_answered": false,
            "view_count": 21,
            "answer_count": 0,
            "score": 0,
            "last_activity_date": 1526415722,
            "creation_date": 1526415722,
            "question_id": 50358612,
            "body_markdown": "My primary concern here is to know how to **launch another instance of my UWP app from the same app itself, some code snippet or documentation link would be helpful**.\r\n\r\nBelow are the changes suggested by different links, but didn&#39;t found any thing concrete:\r\n\r\n**Appraoch:**\r\n\r\nResearch suggest that we can launch multiple instance of the same UWP application by setting the `package.appxmanifest` file with flag `SupportsMultiInstances` to `True`. [Link here][1]\r\n\r\n    &lt;Application Id=&quot;App&quot;\r\n      ...\r\n      desktop4:SupportsMultipleInstances=&quot;true&quot;\r\n      iot2:SupportsMultipleInstances=&quot;true&quot;&gt;\r\n      ...\r\n    &lt;/Application&gt;\r\n\r\n**Extension**\r\n\r\nWe can set the `uap:FileTypeAssociation` for these apps and later run the instance of the app which we wish.  But these are already build files and I need to open new instance at runtime from my uwp app.\r\n\r\n**URI:**\r\n\r\nOne more way was to open store app using URI.\r\n\r\n    var uriToLaunch = &quot;testapp-mainpage://&quot;;\r\n    var uri = new Uri(uriToLaunch);\r\n    bool success = await Windows.System.Launcher.LaunchUriAsync(uri);\r\n\r\n\r\nNow based on this, the way I wish to launch second or another instance of my app is something like below:\r\n\r\nSuppose I have an UWP app which have first view to select type of user from dropdown having Agent, Owner and GuestUser. based on that I show list of House properties to user on next page. \r\n\r\nNow I would like to have the functionality where, when user press and hold on a house property card then the details to purchase it and other booking page(s) are visible on another instance(where selecting the type of user and selecting the house property is skipped). while if user just click once (not long-press) on a house property card then it will open in the same instance.\r\n\r\n**SideNote**\r\n\r\nI am not looking for [MultipleView][2] where we show independent part of your app in separate window.\r\n\r\n\r\n  [1]: https://docs.microsoft.com/en-us/windows/uwp/launch-resume/multi-instance-uwp\r\n  [2]: https://docs.microsoft.com/en-us/windows/uwp/design/layout/show-multiple-views",
            "link": "https://stackoverflow.com/questions/50358612/launch-one-more-instance-of-same-uwp-app-from-first-app",
            "title": "Launch one more instance of same UWP app from first app",
            "body": "<p>My primary concern here is to know how to <strong>launch another instance of my UWP app from the same app itself, some code snippet or documentation link would be helpful</strong>.</p>\n\n<p>Below are the changes suggested by different links, but didn't found any thing concrete:</p>\n\n<p><strong>Appraoch:</strong></p>\n\n<p>Research suggest that we can launch multiple instance of the same UWP application by setting the <code>package.appxmanifest</code> file with flag <code>SupportsMultiInstances</code> to <code>True</code>. <a href=\"https://docs.microsoft.com/en-us/windows/uwp/launch-resume/multi-instance-uwp\" rel=\"nofollow noreferrer\">Link here</a></p>\n\n<pre><code>&lt;Application Id=\"App\"\n  ...\n  desktop4:SupportsMultipleInstances=\"true\"\n  iot2:SupportsMultipleInstances=\"true\"&gt;\n  ...\n&lt;/Application&gt;\n</code></pre>\n\n<p><strong>Extension</strong></p>\n\n<p>We can set the <code>uap:FileTypeAssociation</code> for these apps and later run the instance of the app which we wish.  But these are already build files and I need to open new instance at runtime from my uwp app.</p>\n\n<p><strong>URI:</strong></p>\n\n<p>One more way was to open store app using URI.</p>\n\n<pre><code>var uriToLaunch = \"testapp-mainpage://\";\nvar uri = new Uri(uriToLaunch);\nbool success = await Windows.System.Launcher.LaunchUriAsync(uri);\n</code></pre>\n\n<p>Now based on this, the way I wish to launch second or another instance of my app is something like below:</p>\n\n<p>Suppose I have an UWP app which have first view to select type of user from dropdown having Agent, Owner and GuestUser. based on that I show list of House properties to user on next page. </p>\n\n<p>Now I would like to have the functionality where, when user press and hold on a house property card then the details to purchase it and other booking page(s) are visible on another instance(where selecting the type of user and selecting the house property is skipped). while if user just click once (not long-press) on a house property card then it will open in the same instance.</p>\n\n<p><strong>SideNote</strong></p>\n\n<p>I am not looking for <a href=\"https://docs.microsoft.com/en-us/windows/uwp/design/layout/show-multiple-views\" rel=\"nofollow noreferrer\">MultipleView</a> where we show independent part of your app in separate window.</p>\n"
        },
        {
            "tags": [
                "soap",
                "coldfusion"
            ],
            "owner": {
                "reputation": 2486,
                "user_id": 8345,
                "user_type": "registered",
                "accept_rate": 68,
                "profile_image": "https://www.gravatar.com/avatar/44fa6fc98d16e06b3cdee8aea363e1cf?s=128&d=identicon&r=PG",
                "display_name": "Dan Roberts",
                "link": "https://stackoverflow.com/users/8345/dan-roberts"
            },
            "is_answered": true,
            "view_count": 49,
            "accepted_answer_id": 50358611,
            "answer_count": 1,
            "score": 0,
            "last_activity_date": 1526415720,
            "creation_date": 1525295502,
            "question_id": 50143626,
            "body_markdown": "We are in the process of migrating from the latest update on CF10 to the latest update on CF2016. An issue we are running into is that for the webservices we publish, when we previously specified returntype as components the WSDL and response XML preserved the camel-casing. After migrating all of the complex return types are lower case. The change was so minor we didn&#39;t even notice it in testing but it is apparently causing some users problems with code written for the original casing.\r\n\r\nI have search around and can not find a mention of a change forcing everything to lowercase but that appears to be what is happening. We are still on Windows Server, going from 2008 to 2012, so no change in OS that might warrant this.\r\n\r\nDoes anyone know when this change went into effect and/or how to tweak settings to preserve casing?",
            "link": "https://stackoverflow.com/questions/50143626/how-to-make-soap-types-not-force-lowercase-after-migrating-from-cf10-to-cf2016",
            "title": "How to make SOAP types not force lowercase after migrating from CF10 to CF2016?",
            "body": "<p>We are in the process of migrating from the latest update on CF10 to the latest update on CF2016. An issue we are running into is that for the webservices we publish, when we previously specified returntype as components the WSDL and response XML preserved the camel-casing. After migrating all of the complex return types are lower case. The change was so minor we didn't even notice it in testing but it is apparently causing some users problems with code written for the original casing.</p>\n\n<p>I have search around and can not find a mention of a change forcing everything to lowercase but that appears to be what is happening. We are still on Windows Server, going from 2008 to 2012, so no change in OS that might warrant this.</p>\n\n<p>Does anyone know when this change went into effect and/or how to tweak settings to preserve casing?</p>\n"
        },
        {
            "tags": [
                "python",
                "tensorflow"
            ],
            "owner": {
                "reputation": 3926,
                "user_id": 5904928,
                "user_type": "registered",
                "accept_rate": 86,
                "profile_image": "https://i.stack.imgur.com/jwFz9.png?s=128&g=1",
                "display_name": "Ayodhyankit Paul",
                "link": "https://stackoverflow.com/users/5904928/ayodhyankit-paul"
            },
            "is_answered": false,
            "view_count": 23,
            "answer_count": 0,
            "score": 0,
            "last_activity_date": 1526415709,
            "creation_date": 1526414507,
            "question_id": 50358345,
            "body_markdown": "I am getting a tensor as argument , Now i want to create some variables , I am trying tf.get_variables ( I don&#39;t want to use tf.Variable )\r\n\r\n    input=tensor_argument\r\n        \r\n        \r\n    sequence_length = tf.shape(inputs)[1]  # the length \r\n        \r\n    hidden_size = tf.shape(inputs)[2] # hidden size \r\n\r\n    W_omega = tf.get_variable(name=&#39;w_omega&#39;,shape=[sequence_length,sequence_length],dtype=tf.float32,initializer=tf.random_uniform_initializer(-0.01,0.01))\r\n\r\n\r\n\r\n\r\nWhen i run this code , I am getting this error\r\n\r\n    TypeError: int() argument must be a string, a bytes-like object or a number, not &#39;Tensor&#39;\r\n\r\n\r\nI tried with tf.shape and tf.size() but nothing is working , \r\n\r\n\r\nExample:\r\n\r\n    import tensorflow as tf\r\n    import numpy as np\r\n    \r\n    \r\n    data=np.random.randint(0,10,[2,4,300])\r\n    \r\n    tensor_va=tf.constant(data)\r\n    \r\n    d=tf.shape(tensor_va)[1]\r\n    W_omega = tf.get_variable(name=&#39;a_omega&#39;, shape=[d,d], dtype=tf.float32,\r\n                              initializer=tf.random_uniform_initializer(-0.01, 0.01))\r\n    \r\n    with tf.Session() as sess:\r\n        sess.run(tf.global_variables_initializer())\r\n        print(sess.run(W_omega))\r\n\r\nHow to use or convert tensor to int for get_variable ?\r\n",
            "link": "https://stackoverflow.com/questions/50358345/tensorflow-typeerror-int-argument-must-be-a-string-a-bytes-like-object-or",
            "title": "Tensorflow : TypeError: int() argument must be a string, a bytes-like object or a number, not &#39;Tensor&#39;",
            "body": "<p>I am getting a tensor as argument , Now i want to create some variables , I am trying tf.get_variables ( I don't want to use tf.Variable )</p>\n\n<pre><code>input=tensor_argument\n\n\nsequence_length = tf.shape(inputs)[1]  # the length \n\nhidden_size = tf.shape(inputs)[2] # hidden size \n\nW_omega = tf.get_variable(name='w_omega',shape=[sequence_length,sequence_length],dtype=tf.float32,initializer=tf.random_uniform_initializer(-0.01,0.01))\n</code></pre>\n\n<p>When i run this code , I am getting this error</p>\n\n<pre><code>TypeError: int() argument must be a string, a bytes-like object or a number, not 'Tensor'\n</code></pre>\n\n<p>I tried with tf.shape and tf.size() but nothing is working , </p>\n\n<p>Example:</p>\n\n<pre><code>import tensorflow as tf\nimport numpy as np\n\n\ndata=np.random.randint(0,10,[2,4,300])\n\ntensor_va=tf.constant(data)\n\nd=tf.shape(tensor_va)[1]\nW_omega = tf.get_variable(name='a_omega', shape=[d,d], dtype=tf.float32,\n                          initializer=tf.random_uniform_initializer(-0.01, 0.01))\n\nwith tf.Session() as sess:\n    sess.run(tf.global_variables_initializer())\n    print(sess.run(W_omega))\n</code></pre>\n\n<p>How to use or convert tensor to int for get_variable ?</p>\n"
        },
        {
            "tags": [
                "javascript",
                "promise",
                "async-await"
            ],
            "owner": {
                "reputation": 424,
                "user_id": 3474412,
                "user_type": "registered",
                "accept_rate": 88,
                "profile_image": "https://www.gravatar.com/avatar/0543e840a7894840d579e57f92f6d846?s=128&d=identicon&r=PG&f=1",
                "display_name": "depiction",
                "link": "https://stackoverflow.com/users/3474412/depiction"
            },
            "is_answered": false,
            "view_count": 30,
            "closed_date": 1526416073,
            "answer_count": 0,
            "score": 0,
            "last_activity_date": 1526415701,
            "creation_date": 1526415087,
            "question_id": 50358464,
            "body_markdown": "I have a function that is returning a null value instead of waiting for the inner promise to resolve before returning.\r\n\r\nI&#39;ve tried using async/await, but wasn&#39;t successful. The getTargets function needs to return upload_url, not a promise.\r\n\r\n\t\tfunction getTargets (currentChunk) {\r\n\t\t\tlet chunkArray = [];\r\n\r\n\t\t\t// split chunk strings into key value pairs, then add them to an array\r\n\t\t\tfor(let parameter of currentChunk) {\r\n\t\t\t\tlet keyValue = parameter.split(&#39;=&#39;);\r\n\t\t\t\tchunkArray[keyValue[0]] = keyValue[1];\r\n\t\t\t}\r\n\r\n\t\t\tfor(let file of myFiles) {\r\n\t\t\t\tif (file.name === chunkArray[&#39;resumableFilename&#39;]) {\r\n\t\t\t\t\tmyService.getUploadUrl(file.id, chunkArray[&#39;resumableChunkNumber&#39;], file.upload_id).then((response) =&gt; {\r\n\t\t\t\t\t\tconsole.log(response.upload_url);\r\n\t\t\t\t\t\treturn response.upload_url;\r\n\t\t\t\t\t});\r\n\t\t\t\t}\r\n\t\t\t}\r\n\t\t}",
            "link": "https://stackoverflow.com/questions/50358464/wait-until-inner-promise-resolves-before-returning",
            "closed_reason": "duplicate",
            "title": "Wait until inner promise resolves before returning",
            "body": "<p>I have a function that is returning a null value instead of waiting for the inner promise to resolve before returning.</p>\n\n<p>I've tried using async/await, but wasn't successful. The getTargets function needs to return upload_url, not a promise.</p>\n\n<pre><code>    function getTargets (currentChunk) {\n        let chunkArray = [];\n\n        // split chunk strings into key value pairs, then add them to an array\n        for(let parameter of currentChunk) {\n            let keyValue = parameter.split('=');\n            chunkArray[keyValue[0]] = keyValue[1];\n        }\n\n        for(let file of myFiles) {\n            if (file.name === chunkArray['resumableFilename']) {\n                myService.getUploadUrl(file.id, chunkArray['resumableChunkNumber'], file.upload_id).then((response) =&gt; {\n                    console.log(response.upload_url);\n                    return response.upload_url;\n                });\n            }\n        }\n    }\n</code></pre>\n"
        },
        {
            "tags": [
                "web-services",
                "jax-ws",
                "weblogic11g",
                "mtom"
            ],
            "owner": {
                "reputation": 61,
                "user_id": 2358905,
                "user_type": "registered",
                "profile_image": "https://www.gravatar.com/avatar/854f84c89ac7162f8843e53420874b5c?s=128&d=identicon&r=PG",
                "display_name": "Rakesh Prajapati",
                "link": "https://stackoverflow.com/users/2358905/rakesh-prajapati"
            },
            "is_answered": false,
            "view_count": 25,
            "answer_count": 1,
            "score": 0,
            "last_activity_date": 1526415696,
            "creation_date": 1512080367,
            "last_edit_date": 1513015665,
            "question_id": 47583578,
            "body_markdown": "I have developed a JAX-WS webservice to deal with large file (upload / download). MTOM Attachment + Streaming feature is enabled on the webservice and the client both. Deployed in Weblogic 10.3.6 JDK 1.7\r\n\r\n    // Enable attachment streaming feature\r\n    // dir - attachment above memoryThreshold are stored in this directory\r\n    // parseEagerly - Streaming attachments are to be parsed eagerly (read or write the complete attachment)\r\n    // memoryThreshold - Memory threshold set to 4MB. Attachments under 4MB are stored in memory.\r\n    @StreamingAttachment(dir=&quot;C:/Users/rakesh/Desktop/temp/server&quot;, parseEagerly=true, memoryThreshold=40000L)\r\n    //Enable MTOM (Message Transmission Optimization Mechanism) feature for xs:binary64 data over 10KB (10240 bytes)\r\n    @MTOM(enabled=true, threshold=10240)\r\n    @WebService\r\n    public class ImagingStreamingWebService {\r\n    ...\r\n    }\r\n\r\nMTOM streaming is working fine in both upload and download which great. Issue is in case of upload when client sending a large attachment bigger than 4MB. Based on the @StreamingAttachment annotation attachment is saved to `dir=&quot;C:/Users/rakesh/Desktop/temp/server&quot;` After processing is done and response is sent back to client, temp attachment file is not deleted from dir above.\r\n\r\nTemp attachment files are not deleted even after JVM recycle (shutdown of server). See the picture of files in temp dir below\r\n\r\n[![Temp attachment files on FileSystem][1]][1]\r\n\r\nI can not find any way to delete these files. Manually deleting or batch/script to cleanup dir is not possible because these are locked by JVM. Ideal is to deleted theses files as soon as webmethod processing is done.\r\n\r\n  [1]: https://i.stack.imgur.com/RU6b6.jpg",
            "link": "https://stackoverflow.com/questions/47583578/weblogic-10-3-6-jax-ws-webservice-temporary-copies-of-mtom-attachments-are-not-b",
            "title": "Weblogic 10.3.6 JAX-WS Webservice Temporary copies of MTOM attachments are not being deleted from FileSystem",
            "body": "<p>I have developed a JAX-WS webservice to deal with large file (upload / download). MTOM Attachment + Streaming feature is enabled on the webservice and the client both. Deployed in Weblogic 10.3.6 JDK 1.7</p>\n\n<pre><code>// Enable attachment streaming feature\n// dir - attachment above memoryThreshold are stored in this directory\n// parseEagerly - Streaming attachments are to be parsed eagerly (read or write the complete attachment)\n// memoryThreshold - Memory threshold set to 4MB. Attachments under 4MB are stored in memory.\n@StreamingAttachment(dir=\"C:/Users/rakesh/Desktop/temp/server\", parseEagerly=true, memoryThreshold=40000L)\n//Enable MTOM (Message Transmission Optimization Mechanism) feature for xs:binary64 data over 10KB (10240 bytes)\n@MTOM(enabled=true, threshold=10240)\n@WebService\npublic class ImagingStreamingWebService {\n...\n}\n</code></pre>\n\n<p>MTOM streaming is working fine in both upload and download which great. Issue is in case of upload when client sending a large attachment bigger than 4MB. Based on the @StreamingAttachment annotation attachment is saved to <code>dir=\"C:/Users/rakesh/Desktop/temp/server\"</code> After processing is done and response is sent back to client, temp attachment file is not deleted from dir above.</p>\n\n<p>Temp attachment files are not deleted even after JVM recycle (shutdown of server). See the picture of files in temp dir below</p>\n\n<p><a href=\"https://i.stack.imgur.com/RU6b6.jpg\" rel=\"nofollow noreferrer\"><img src=\"https://i.stack.imgur.com/RU6b6.jpg\" alt=\"Temp attachment files on FileSystem\"></a></p>\n\n<p>I can not find any way to delete these files. Manually deleting or batch/script to cleanup dir is not possible because these are locked by JVM. Ideal is to deleted theses files as soon as webmethod processing is done.</p>\n"
        }
    ],
    "has_more": true,
    "quota_max": 300,
    "quota_remaining": 299
}